diff -Naur --exclude=.git tcl/unix/configure TCL_BLD/unix/configure
--- tcl/unix/configure	2026-09-10 16:55:00.590363000 +0000
+++ TCL_BLD/unix/configure	2026-09-10 17:00:55.104920000 +0000
@@ -5992,13 +5992,13 @@
 
 if test $zlib_ok = yes; then
 
-  echo "$as_me:$LINENO: checking for library containing deflateSetHeader" >&5
-echo $ECHO_N "checking for library containing deflateSetHeader... $ECHO_C" >&6
-if test "${ac_cv_search_deflateSetHeader+set}" = set; then
+  echo "$as_me:$LINENO: checking for library containing brl_deflateSetHeader" >&5
+echo $ECHO_N "checking for library containing brl_deflateSetHeader... $ECHO_C" >&6
+if test "${ac_cv_search_brl_deflateSetHeader+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_func_search_save_LIBS=$LIBS
-ac_cv_search_deflateSetHeader=no
+ac_cv_search_brl_deflateSetHeader=no
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -6012,11 +6012,11 @@
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char deflateSetHeader ();
+char brl_deflateSetHeader ();
 int
 main ()
 {
-deflateSetHeader ();
+brl_deflateSetHeader ();
   ;
   return 0;
 }
@@ -6043,7 +6043,7 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_search_deflateSetHeader="none required"
+  ac_cv_search_brl_deflateSetHeader="none required"
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
@@ -6051,8 +6051,8 @@
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
-if test "$ac_cv_search_deflateSetHeader" = no; then
-  for ac_lib in z; do
+if test "$ac_cv_search_brl_deflateSetHeader" = no; then
+  for ac_lib in z_brl; do
     LIBS="-l$ac_lib  $ac_func_search_save_LIBS"
     cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -6067,11 +6067,11 @@
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char deflateSetHeader ();
+char brl_deflateSetHeader ();
 int
 main ()
 {
-deflateSetHeader ();
+brl_deflateSetHeader ();
   ;
   return 0;
 }
@@ -6098,7 +6098,7 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_search_deflateSetHeader="-l$ac_lib"
+  ac_cv_search_brl_deflateSetHeader="-l$ac_lib"
 break
 else
   echo "$as_me: failed program was:" >&5
@@ -6111,10 +6111,10 @@
 fi
 LIBS=$ac_func_search_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_search_deflateSetHeader" >&5
-echo "${ECHO_T}$ac_cv_search_deflateSetHeader" >&6
-if test "$ac_cv_search_deflateSetHeader" != no; then
-  test "$ac_cv_search_deflateSetHeader" = "none required" || LIBS="$ac_cv_search_deflateSetHeader $LIBS"
+echo "$as_me:$LINENO: result: $ac_cv_search_brl_deflateSetHeader" >&5
+echo "${ECHO_T}$ac_cv_search_brl_deflateSetHeader" >&6
+if test "$ac_cv_search_brl_deflateSetHeader" != no; then
+  test "$ac_cv_search_brl_deflateSetHeader" = "none required" || LIBS="$ac_cv_search_brl_deflateSetHeader $LIBS"
 
 else
 
diff -Naur --exclude=.git tcl/unix/configure.in TCL_BLD/unix/configure.in
--- tcl/unix/configure.in	2026-09-10 16:55:00.590431000 +0000
+++ TCL_BLD/unix/configure.in	2026-09-10 17:00:55.107077000 +0000
@@ -163,7 +163,7 @@
   AC_CHECK_TYPE([gz_header],[],[zlib_ok=no],[#include <zlib.h>])],[
   zlib_ok=no])
 AS_IF([test $zlib_ok = yes], [
-  AC_SEARCH_LIBS([deflateSetHeader],[z],[],[
+  AC_SEARCH_LIBS([brl_deflateSetHeader],[z_brl],[],[
     zlib_ok=no
   ])])
 AS_IF([test $zlib_ok = no], [
