diff -Naur --exclude=.git TCL_BLD/macosx/GNUmakefile TCL_BLD/macosx/GNUmakefile
--- TCL_BLD/macosx/GNUmakefile	2026-09-10 17:00:41.376647000 +0000
+++ TCL_BLD/macosx/GNUmakefile	2026-09-10 17:19:13.741872000 +0000
@@ -117,10 +117,7 @@
 INSTALL_TARGETS		+= install-private-headers
 endif
 ifeq (${INSTALL_BUILD}_${EMBEDDED_BUILD}_${BUILD_STYLE},1__Deployment)
-INSTALL_TARGETS		+= install-packages html-tcl
-ifneq (${INSTALL_MANPAGES},)
-INSTALL_TARGETS		+= install-doc
-endif
+INSTALL_TARGETS		+= install-packages
 endif
 
 MAKE_VARS		:= INSTALL_ROOT INSTALL_TARGETS VERSION GENERIC_FLAGS
diff -Naur --exclude=.git TCL_BLD/unix/Makefile.in TCL_BLD/unix/Makefile.in
--- TCL_BLD/unix/Makefile.in	2026-09-10 17:09:34.696388000 +0000
+++ TCL_BLD/unix/Makefile.in	2026-09-10 17:19:13.743997000 +0000
@@ -622,7 +622,7 @@
 # Start of rules
 #--------------------------------------------------------------------------
 
-all: binaries libraries doc packages
+all: binaries libraries packages
 
 binaries: ${LIB_FILE} ${TCL_EXE}
 
@@ -792,7 +792,7 @@
 INSTALL_PACKAGE_TARGETS = install-packages
 INSTALL_DEV_TARGETS = install-headers
 INSTALL_EXTRA_TARGETS = @EXTRA_INSTALL@
-INSTALL_TARGETS = $(INSTALL_BASE_TARGETS) $(INSTALL_DEV_TARGETS) $(INSTALL_DOC_TARGETS) \
+INSTALL_TARGETS = $(INSTALL_BASE_TARGETS) $(INSTALL_DEV_TARGETS) \
 		  $(INSTALL_PACKAGE_TARGETS) $(INSTALL_EXTRA_TARGETS)
 
 install: $(INSTALL_TARGETS)
diff -Naur --exclude=.git TCL_BLD/unix/configure TCL_BLD/unix/configure
--- TCL_BLD/unix/configure	2026-09-10 17:09:34.687269000 +0000
+++ TCL_BLD/unix/configure	2026-09-10 17:19:13.760019000 +0000
@@ -19201,7 +19201,7 @@
     includedir="${libdir}/Headers"
     PRIVATE_INCLUDE_DIR="${libdir}/PrivateHeaders"
     HTML_DIR="${libdir}/Resources/Documentation/Reference/Tcl"
-    EXTRA_INSTALL="install-private-headers html-tcl"
+    EXTRA_INSTALL="install-private-headers"
     EXTRA_BUILD_HTML='@ln -fs contents.htm "$(HTML_INSTALL_DIR)/TclTOC.html"'
     EXTRA_INSTALL_BINARIES='@echo "Installing Info.plist to $(LIB_INSTALL_DIR)/Resources/" && $(INSTALL_DATA_DIR) "$(LIB_INSTALL_DIR)/Resources" && $(INSTALL_DATA) Tcl-Info.plist "$(LIB_INSTALL_DIR)/Resources/Info.plist"'
     EXTRA_INSTALL_BINARIES="$EXTRA_INSTALL_BINARIES"' && echo "Installing license.terms to $(LIB_INSTALL_DIR)/Resources/" && $(INSTALL_DATA) "$(TOP_DIR)/license.terms" "$(LIB_INSTALL_DIR)/Resources"'
diff -Naur --exclude=.git TCL_BLD/unix/configure.in TCL_BLD/unix/configure.in
--- TCL_BLD/unix/configure.in	2026-09-10 17:09:34.687465000 +0000
+++ TCL_BLD/unix/configure.in	2026-09-10 17:19:13.762242000 +0000
@@ -826,7 +826,7 @@
     includedir="${libdir}/Headers"
     PRIVATE_INCLUDE_DIR="${libdir}/PrivateHeaders"
     HTML_DIR="${libdir}/Resources/Documentation/Reference/Tcl"
-    EXTRA_INSTALL="install-private-headers html-tcl"
+    EXTRA_INSTALL="install-private-headers"
     EXTRA_BUILD_HTML='@ln -fs contents.htm "$(HTML_INSTALL_DIR)/TclTOC.html"'
     EXTRA_INSTALL_BINARIES='@echo "Installing Info.plist to $(LIB_INSTALL_DIR)/Resources/" && $(INSTALL_DATA_DIR) "$(LIB_INSTALL_DIR)/Resources" && $(INSTALL_DATA) Tcl-Info.plist "$(LIB_INSTALL_DIR)/Resources/Info.plist"'
     EXTRA_INSTALL_BINARIES="$EXTRA_INSTALL_BINARIES"' && echo "Installing license.terms to $(LIB_INSTALL_DIR)/Resources/" && $(INSTALL_DATA) "$(TOP_DIR)/license.terms" "$(LIB_INSTALL_DIR)/Resources"'
diff -Naur --exclude=.git TCL_BLD/win/Makefile.in TCL_BLD/win/Makefile.in
--- TCL_BLD/win/Makefile.in	2026-09-10 17:10:10.333775000 +0000
+++ TCL_BLD/win/Makefile.in	2026-09-10 17:19:13.763317000 +0000
@@ -441,7 +441,7 @@
 
 TCL_DOCS = "$(ROOT_DIR_NATIVE)"/doc/*.[13n]
 
-all: binaries libraries doc packages
+all: binaries libraries packages
 
 # Test-suite helper (can be used to test Tcl from build directory with all expected modules).
 # To start from windows shell use:
@@ -658,7 +658,7 @@
 		"$(TOMMATH_DIR_NATIVE)/tommath.h" \
 		> "$(GENERIC_DIR_NATIVE)/tclTomMath.h"
 
-install: all install-binaries install-libraries install-headers install-doc install-packages
+install: all install-binaries install-libraries install-headers install-packages
 
 install-binaries: binaries
 	@for i in "$(LIB_INSTALL_DIR)" "$(BIN_INSTALL_DIR)"; \
diff -Naur --exclude=.git TCL_BLD/win/makefile.vc TCL_BLD/win/makefile.vc
--- TCL_BLD/win/makefile.vc	2026-09-10 17:10:32.280973000 +0000
+++ TCL_BLD/win/makefile.vc	2026-09-10 17:19:13.765572000 +0000
@@ -468,7 +468,7 @@
 shell:	    setup $(TCLSH)
 dlls:	    setup $(TCLREGLIB) $(TCLDDELIB)
 tcltest:    setup $(TCLTEST) dlls $(CAT32)
-install:    install-binaries install-libraries install-docs install-pkgs
+install:    install-binaries install-libraries install-pkgs
 !if $(SYMBOLS)
 install:    install-pdbs
 !endif
