diff -Naur libexpat/expat/CMakeLists.txt EXPAT_BLD/expat/CMakeLists.txt
--- libexpat/expat/CMakeLists.txt	2024-06-24 20:14:05.745779978 -0400
+++ EXPAT_BLD/expat/CMakeLists.txt	2024-06-24 20:13:21.062107763 -0400
@@ -462,7 +462,7 @@
 
     expat_shy_set(EXPAT_${build_type_upper}_POSTFIX "${_POSTFIX_WIDE}${_POSTFIX_DEBUG}${_POSTFIX_CRT}" CACHE STRING "Library filename postfix for build type ${build_type_upper}; yields filenames libexpat<postfix=[w][d][MD|MT]>.(dll|dylib|lib|so)")
     mark_as_advanced(EXPAT_${build_type_upper}_POSTFIX)
-    set_property(TARGET expat PROPERTY ${build_type_upper}_POSTFIX ${EXPAT_${build_type_upper}_POSTFIX})
+	 #set_property(TARGET expat PROPERTY ${build_type_upper}_POSTFIX ${EXPAT_${build_type_upper}_POSTFIX})
 endforeach()
 
 set(LIBCURRENT 10)  # sync
