--- assimp/CMakeLists.txt	2025-07-22 16:48:44.626337172 -0400
+++ ASSETIMPORT_BLD/CMakeLists.txt	2025-07-22 16:28:32.943270150 -0400
@@ -37,6 +37,7 @@
 SET(CMAKE_POLICY_DEFAULT_CMP0012 NEW)
 SET(CMAKE_POLICY_DEFAULT_CMP0074 NEW)
 SET(CMAKE_POLICY_DEFAULT_CMP0092 NEW)
+SET(CMAKE_POLICY_DEFAULT_CMP0144 NEW)
 
 CMAKE_MINIMUM_REQUIRED( VERSION 3.22 )
 
--- assimp/cmake-modules/assimp-plain-config.cmake.in	2025-07-22 16:48:43.776340530 -0400
+++ ASSETIMPORT_BLD/cmake-modules/assimp-plain-config.cmake.in	2025-07-22 16:28:33.458267904 -0400
@@ -1,5 +1,7 @@
 @PACKAGE_INIT@
 
+find_package(ZLIB MODULE REQUIRED)
+
 include("${CMAKE_CURRENT_LIST_DIR}/@TARGETS_EXPORT_NAME@.cmake")
 
 set(ASSIMP_ROOT_DIR ${PACKAGE_PREFIX_DIR})
diff -Naur assimp/code/CMakeLists.txt ASSETIMPORT_BLD/code/CMakeLists.txt
--- assimp/code/CMakeLists.txt	2026-02-04 16:56:22.592686074 -0500
+++ ASSETIMPORT_BLD/code/CMakeLists.txt	2026-02-04 16:55:19.040925630 -0500
@@ -1113,18 +1113,19 @@
 ENDIF() # IF (ASSIMP_BUILD_USD_IMPORTER)
 
 # pugixml
-IF(ASSIMP_HUNTER_ENABLED)
-  hunter_add_package(pugixml)
-  find_package(pugixml CONFIG REQUIRED)
-ELSEIF(NOT TARGET pugixml::pugixml)
-  SET( Pugixml_SRCS
-    ../contrib/pugixml/src/pugiconfig.hpp
-    ../contrib/pugixml/src/pugixml.cpp
-    ../contrib/pugixml/src/pugixml.hpp
-  )
-  INCLUDE_DIRECTORIES("../contrib/pugixml/src")
-  SOURCE_GROUP( Contrib\\Pugixml FILES ${Pugixml_SRCS})
-ENDIF()
+#IF(ASSIMP_HUNTER_ENABLED)
+#  hunter_add_package(pugixml)
+#  find_package(pugixml CONFIG REQUIRED)
+#ELSEIF(NOT TARGET pugixml::pugixml)
+#  SET( Pugixml_SRCS
+#    ../contrib/pugixml/src/pugiconfig.hpp
+#    ../contrib/pugixml/src/pugixml.cpp
+#    ../contrib/pugixml/src/pugixml.hpp
+#  )
+#  INCLUDE_DIRECTORIES("../contrib/pugixml/src")
+#  SOURCE_GROUP( Contrib\\Pugixml FILES ${Pugixml_SRCS})
+#ENDIF()
+find_package(pugixml REQUIRED)
 
 # utf8
 IF(ASSIMP_HUNTER_ENABLED)
@@ -1414,6 +1415,10 @@
   $<INSTALL_INTERFACE:${ASSIMP_INCLUDE_INSTALL_DIR}>
 )
 
+if (TARGET pugixml)
+	set(PUGIXML_LIBRARIES pugixml)
+endif (TARGET pugixml)
+
 IF(ASSIMP_HUNTER_ENABLED)
   TARGET_LINK_LIBRARIES(assimp
       PUBLIC
@@ -1435,7 +1440,8 @@
     target_link_libraries(assimp PRIVATE ${draco_LIBRARIES})
   endif()
 ELSE()
-  TARGET_LINK_LIBRARIES(assimp ${ZLIB_LIBRARIES} ${OPENDDL_PARSER_LIBRARIES})
+  find_package(ZLIB MODULE REQUIRED)
+  TARGET_LINK_LIBRARIES(assimp ZLIB::ZLIB ${PUGIXML_LIBRARIES} ${OPENDDL_PARSER_LIBRARIES})
   if (ASSIMP_BUILD_DRACO)
     target_link_libraries(assimp ${draco_LIBRARIES})
   endif()
@@ -1453,6 +1459,7 @@
   TARGET_LINK_LIBRARIES(assimp ${C4D_EXTRA_LIBRARIES})
 ENDIF ()
 
+if(0)
 if( MSVC )
   # in order to prevent DLL hell, each of the DLLs have to be suffixed with the major version and msvc prefix
   # CMake 3.12 added a variable for this
@@ -1499,11 +1506,12 @@
     target_compile_definitions(assimp PUBLIC WindowsStore)
     TARGET_LINK_LIBRARIES(assimp advapi32)
 endif()
+endif(0)
 
 SET_TARGET_PROPERTIES( assimp PROPERTIES
   VERSION ${ASSIMP_VERSION}
   SOVERSION ${ASSIMP_SOVERSION} # use full version
-  OUTPUT_NAME assimp${LIBRARY_SUFFIX}
+  #OUTPUT_NAME assimp${LIBRARY_SUFFIX}
 )
 
 if (APPLE)
