diff -Naur assimp/CMakeLists.txt ASSETIMPORT_BLD/CMakeLists.txt
--- assimp/CMakeLists.txt	2024-10-11 09:42:55.176363481 -0400
+++ ASSETIMPORT_BLD/CMakeLists.txt	2024-10-11 09:46:32.046826530 -0400
@@ -37,6 +37,7 @@
 SET(CMAKE_POLICY_DEFAULT_CMP0012 NEW)
 SET(CMAKE_POLICY_DEFAULT_CMP0074 NEW)
 SET(CMAKE_POLICY_DEFAULT_CMP0092 NEW)
+SET(CMAKE_POLICY_DEFAULT_CMP0144 NEW)
 
 CMAKE_MINIMUM_REQUIRED( VERSION 3.22 )
 
@@ -506,10 +507,10 @@
   set(ASSIMP_BUILD_MINIZIP TRUE)
 ELSE()
   # If the zlib is already found outside, add an export in case assimpTargets can't find it.
-  IF( ZLIB_FOUND AND ASSIMP_INSTALL)
-    INSTALL( TARGETS zlib zlibstatic
-        EXPORT "${TARGETS_EXPORT_NAME}")
-  ENDIF()
+  #IF( ZLIB_FOUND AND ASSIMP_INSTALL)
+  #  INSTALL( TARGETS zlib zlibstatic
+  #      EXPORT "${TARGETS_EXPORT_NAME}")
+  #ENDIF()
 
   IF ( NOT ASSIMP_BUILD_ZLIB )
     FIND_PACKAGE(ZLIB)
@@ -542,9 +543,7 @@
     SET(ASSIMP_BUILD_MINIZIP 1)
   ELSE()
     ADD_DEFINITIONS(-DASSIMP_BUILD_NO_OWN_ZLIB)
-    SET(ZLIB_LIBRARIES_LINKED -lz)
   ENDIF()
-  INCLUDE_DIRECTORIES(${ZLIB_INCLUDE_DIR})
 ENDIF()
 
 IF( NOT IOS )
diff -Naur assimp/code/CMakeLists.txt ASSETIMPORT_BLD/code/CMakeLists.txt
--- assimp/code/CMakeLists.txt	2024-10-11 09:42:55.177363459 -0400
+++ ASSETIMPORT_BLD/code/CMakeLists.txt	2024-10-11 09:47:03.945183936 -0400
@@ -1050,17 +1050,18 @@
 ENDIF() # IF (ASSIMP_BUILD_USD_IMPORTER)
 
 # pugixml
-IF(ASSIMP_HUNTER_ENABLED)
-  hunter_add_package(pugixml)
-  find_package(pugixml CONFIG REQUIRED)
-ELSE()
-  SET( Pugixml_SRCS
-    ../contrib/pugixml/src/pugiconfig.hpp
-    ../contrib/pugixml/src/pugixml.hpp
-  )
-  INCLUDE_DIRECTORIES("../contrib/pugixml/src")
-  SOURCE_GROUP( Contrib\\Pugixml FILES ${Pugixml_SRCS})
-ENDIF()
+#IF(ASSIMP_HUNTER_ENABLED)
+#  hunter_add_package(pugixml)
+#  find_package(pugixml CONFIG REQUIRED)
+#ELSE()
+#  SET( Pugixml_SRCS
+#    ../contrib/pugixml/src/pugiconfig.hpp
+#    ../contrib/pugixml/src/pugixml.hpp
+#  )
+#  INCLUDE_DIRECTORIES("../contrib/pugixml/src")
+#  SOURCE_GROUP( Contrib\\Pugixml FILES ${Pugixml_SRCS})
+#ENDIF()
+find_package(pugixml REQUIRED)
 
 # utf8
 IF(ASSIMP_HUNTER_ENABLED)
@@ -1414,6 +1415,10 @@
   $<INSTALL_INTERFACE:${ASSIMP_INCLUDE_INSTALL_DIR}>
 )
 
+if (TARGET pugixml)
+	set(PUGIXML_LIBRARIES pugixml)
+endif (TARGET pugixml)
+
 IF(ASSIMP_HUNTER_ENABLED)
   TARGET_LINK_LIBRARIES(assimp
       PUBLIC
@@ -1435,7 +1440,7 @@
     target_link_libraries(assimp PRIVATE ${draco_LIBRARIES})
   endif()
 ELSE()
-  TARGET_LINK_LIBRARIES(assimp ${ZLIB_LIBRARIES} ${OPENDDL_PARSER_LIBRARIES})
+  TARGET_LINK_LIBRARIES(assimp ZLIB::ZLIB ${PUGIXML_LIBRARIES} ${OPENDDL_PARSER_LIBRARIES})
   if (ASSIMP_BUILD_DRACO)
     target_link_libraries(assimp ${draco_LIBRARIES})
   endif()
@@ -1453,6 +1458,7 @@
   TARGET_LINK_LIBRARIES(assimp ${C4D_EXTRA_LIBRARIES})
 ENDIF ()
 
+if(0)
 if( MSVC )
   # in order to prevent DLL hell, each of the DLLs have to be suffixed with the major version and msvc prefix
   # CMake 3.12 added a variable for this
@@ -1499,11 +1505,12 @@
     target_compile_definitions(assimp PUBLIC WindowsStore)
     TARGET_LINK_LIBRARIES(assimp advapi32)
 endif()
+endif(0)
 
 SET_TARGET_PROPERTIES( assimp PROPERTIES
   VERSION ${ASSIMP_VERSION}
   SOVERSION ${ASSIMP_SOVERSION} # use full version
-  OUTPUT_NAME assimp${LIBRARY_SUFFIX}
+  #OUTPUT_NAME assimp${LIBRARY_SUFFIX}
 )
 
 if (APPLE)
