diff -Naur assimp/contrib/stb/stb_image.h ASSETIMPORT_BLD/contrib/stb/stb_image.h
--- assimp/contrib/stb/stb_image.h	2026-02-05 14:22:20.343920277 -0500
+++ ASSETIMPORT_BLD/contrib/stb/stb_image.h	2026-02-05 14:24:43.112400148 -0500
@@ -5079,7 +5079,7 @@
 {
    stbi_uc palette[1024], pal_img_n=0;
    stbi_uc has_trans=0, tc[3]={0};
-   stbi__uint16 tc16[3];
+   stbi__uint16 tc16[3]={0};
    stbi__uint32 ioff=0, idata_limit=0, i, pal_len=0;
    int first=1,k,interlace=0, color=0, is_iphone=0;
    stbi__context *s = z->s;
