diff -Naur manifold/src/third_party/thrust/dependencies/libcudacxx/include/cuda/std/detail/libcxx/include/__config MANIFOLD_BLD/src/third_party/thrust/dependencies/libcudacxx/include/cuda/std/detail/libcxx/include/__config
--- manifold/src/third_party/thrust/dependencies/libcudacxx/include/cuda/std/detail/libcxx/include/__config	2026-02-06 16:37:54.212137501 -0500
+++ MANIFOLD_BLD/src/third_party/thrust/dependencies/libcudacxx/include/cuda/std/detail/libcxx/include/__config	2026-02-06 16:37:30.245235005 -0500
@@ -518,9 +518,7 @@
 #define _LIBCUDACXX_HAS_NO_VARIABLE_TEMPLATES
 #endif
 
-#if !(__has_feature(cxx_noexcept))
 #define _LIBCUDACXX_HAS_NO_NOEXCEPT
-#endif
 
 #if __has_feature(underlying_type)
 #define _LIBCUDACXX_UNDERLYING_TYPE(...) __underlying_type(__VA_ARGS__)
