diff -Naur manifold/src/utilities/include/optional_assert.h MANIFOLD_BLD/src/utilities/include/optional_assert.h
--- manifold/src/utilities/include/optional_assert.h	2024-07-02 20:09:19.078438473 -0400
+++ MANIFOLD_BLD/src/utilities/include/optional_assert.h	2024-07-09 16:59:17.609083445 -0400
@@ -15,7 +15,7 @@
 #pragma once
 
 #include "public.h"
-#if MANIFOLD_EXCEPTIONS
+#ifdef MANIFOLD_EXCEPTIONS
 #include <stdexcept>
 #endif
 
