diff -Naur manifold/src/third_party/thrust/thrust/type_traits/is_contiguous_iterator.h MANIFOLD_BLD/src/third_party/thrust/thrust/type_traits/is_contiguous_iterator.h
--- manifold/src/third_party/thrust/thrust/type_traits/is_contiguous_iterator.h	2026-02-06 08:39:38.850209233 -0500
+++ MANIFOLD_BLD/src/third_party/thrust/thrust/type_traits/is_contiguous_iterator.h	2026-02-06 08:39:20.153291216 -0500
@@ -139,7 +139,7 @@
 #if defined(_LIBCPP_VERSION)
 template <typename Iterator>
 struct is_libcxx_wrap_iter<
-  _VSTD::__wrap_iter<Iterator>
+  std::__wrap_iter<Iterator>
 > : true_type {};
 #endif
 
