--- opennurbs/opennurbs_system.h	2024-09-24 11:00:03.443753734 -0400
+++ OPENNURBS_BLD/opennurbs_system.h	2024-09-24 11:04:55.358131669 -0400
@@ -494,7 +494,7 @@
 #pragma ON_PRAGMA_WARNING_AFTER_DIRTY_INCLUDE
 
 #pragma ON_PRAGMA_WARNING_BEFORE_DIRTY_INCLUDE
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM) || defined(ON_RUNTIME_BSD)
 #include "android_uuid/uuid.h"
 #else
 #include <uuid/uuid.h>
