--- opennurbs/opennurbs_defines.h	2024-06-25 15:28:09.119849263 -0400
+++ OPENNURBS_BLD/opennurbs_defines.h	2024-06-25 15:26:59.823993359 -0400
@@ -102,6 +102,11 @@
 #define ON_EXTERN_DECL __declspec(dllimport)
 #define ON_DLL_TEMPLATE extern
 
+#elif defined(__GNUC__)
+#define ON_CLASS __attribute__ ((visibility ("default")))
+#define ON_DECL __attribute__ ((visibility ("default")))
+#define ON_EXTERN_DECL __attribute__ ((visibility ("default")))
+
 #elif defined(ON_COMPILER_CLANG)
 /* using opennurbs as an Apple shared library */
 #define ON_CLASS __attribute__ ((visibility ("default")))
@@ -2351,7 +2356,7 @@
   // the values.  The reason for the gaps between the enum
   // values is to leave room for future snaps with prededence
   // falling between existing snaps
-  enum osnap_mode
+  enum osnap_mode: unsigned int
   {
     os_none          =          0,
     os_near          =          2,
@@ -2741,7 +2746,7 @@
   // Do not change these values; they are stored in 3dm archives
   // and provide a persistent way to identify components of
   // complex objects.
-  enum TYPE
+  enum TYPE: unsigned int
   {
     invalid_type       =   0,
 
