--- opennurbs/opennurbs_zlib.cpp	2024-06-25 15:44:35.958646126 -0400
+++ OPENNURBS_BLD/opennurbs_zlib.cpp	2024-06-25 15:44:06.155134199 -0400
@@ -23,6 +23,7 @@
 
 #include "opennurbs_zlib.h"
 
+#if 0
 #if defined(ON_COMPILER_MSC) && !defined(ON_CMAKE_BUILD)
 
 #if !defined(OPENNURBS_ZLIB_LIB_DIR)
@@ -52,6 +53,8 @@
 
 #endif
 
+#endif
+
 // compressed buffer I/O uses zlib 1.1.3 inflate()/deflate()
 class ON_CompressorImplementation
 {
@@ -319,11 +322,11 @@
       // no uncompressed input is left - switch to finish mode
       flush = Z_FINISH;
     }
-    zrc = z_deflate(&m_zlib.m_strm, flush);
+    zrc = deflate(&m_zlib.m_strm, flush);
     if ( zrc < 0 ) 
     {
       // Something went haywire - bail out.
-      ON_ERROR("ON_BinaryArchive::WriteDeflate - z_deflate failure");
+      ON_ERROR("ON_BinaryArchive::WriteDeflate - deflate failure");
       rc = false;
       break;
     }
@@ -519,11 +522,11 @@
       // no compressed input is left - switch to finish mode
       flush = Z_FINISH;
     }
-    zrc = z_inflate( &m_zlib.m_strm, flush );
+    zrc = inflate( &m_zlib.m_strm, flush );
     if ( zrc < 0 ) 
     {
       // Something went haywire - bail out.
-      ON_ERROR("ON_BinaryArchive::ReadInflate - z_inflate failure");
+      ON_ERROR("ON_BinaryArchive::ReadInflate - inflate failure");
       rc = false;
       break;
     }
@@ -1200,11 +1203,11 @@
       // no uncompressed input is left - switch to finish mode
       flush = Z_FINISH;
     }
-    zrc = z_deflate( &m_zlib.m_strm, flush ); 
+    zrc = deflate( &m_zlib.m_strm, flush ); 
     if ( zrc < 0 ) 
     {
       // Something went haywire - bail out.
-      ON_ERROR("ON_CompressedBuffer::DeflateHelper - z_deflate failure");
+      ON_ERROR("ON_CompressedBuffer::DeflateHelper - deflate failure");
       rc = false;
       break;
     }
@@ -1333,11 +1336,11 @@
       // no compressed input is left - switch to finish mode
       flush = Z_FINISH;
     }
-    zrc = z_inflate( &m_zlib.m_strm, flush );
+    zrc = inflate( &m_zlib.m_strm, flush );
     if ( zrc < 0 ) 
     {
       // Something went haywire - bail out.
-      ON_ERROR("ON_CompressedBuffer::InflateHelper - z_inflate failure");
+      ON_ERROR("ON_CompressedBuffer::InflateHelper - inflate failure");
       rc = false;
       break;
     }
--- opennurbs/opennurbs_zlib.h	2024-06-25 15:44:35.958646126 -0400
+++ OPENNURBS_BLD/opennurbs_zlib.h	2024-06-25 15:44:06.156134182 -0400
@@ -25,19 +25,19 @@
 // and statically link with the zlib library. All the necessary
 // header files are included by opennurbs.h.
 
-
+#if 0
 #if !defined(Z_PREFIX)
 /* decorates zlib functions with a "z_" prefix to prevent symbol collision. */
 #define Z_PREFIX
 #endif
-
+#endif
 #if !defined(MY_ZCALLOC)
 /* have zlib use oncalloc() and onfree() for memory management*/
 #define MY_ZCALLOC
 #endif
 
 #pragma ON_PRAGMA_WARNING_BEFORE_DIRTY_INCLUDE
-#include "./zlib/zlib.h"
+#include "zlib.h"
 #pragma ON_PRAGMA_WARNING_AFTER_DIRTY_INCLUDE
 
 ON_BEGIN_EXTERNC
