--- opennurbs/opennurbs_brep.cpp	2023-12-20 11:23:16.137987286 -0500
+++ OPENNURBS_BLD/opennurbs_brep.cpp	2023-12-20 11:22:15.331051837 -0500
@@ -3578,6 +3578,10 @@
     }   
   }
 
+  // BRL-CAD uses this for a tolerance check below
+  const ON_Surface *surf = loop.Face()->SurfaceOf();
+  double urange = (surf) ? (surf->Domain(0)[1] - surf->Domain(0)[0]) : DBL_MAX;
+  double vrange = (surf) ? (surf->Domain(1)[1] - surf->Domain(1)[0]) : DBL_MAX;
 
   // make sure ends of trims jibe
   int ci0, ci1, next_lti;
@@ -3606,6 +3610,12 @@
       //    didn't get flagged as bad.
       double xtol = (fabs(P0.x) + fabs(P1.x))*1.0e-10;
       double ytol = (fabs(P0.y) + fabs(P1.y))*1.0e-10;
+
+      // Oct 12 2009 Rather than using the above check, BRL-CAD uses
+      // relative uv size if it is available
+      xtol = (urange < DBL_MAX) ? urange * trim0.m_tolerance[0] : xtol;
+      ytol = (vrange < DBL_MAX) ? vrange * trim0.m_tolerance[1] : ytol;
+
       if ( xtol < ON_ZERO_TOLERANCE )
         xtol = ON_ZERO_TOLERANCE;
       if ( ytol < ON_ZERO_TOLERANCE )
