--- opennurbs/opennurbs_brep.h	2024-02-16 13:44:43.176776155 -0500
+++ OPENNURBS_BLD/opennurbs_brep.h	2024-02-16 14:00:20.173055503 -0500
@@ -4141,6 +4141,7 @@
         );
 
   // helpers for validation checking
+public:
   bool IsValidTrim(int trim_index,ON_TextLog* text_log) const;
   bool IsValidTrimTopology(int trim_index,ON_TextLog* text_log) const;
   bool IsValidTrimGeometry(int trim_index,ON_TextLog* text_log) const;
@@ -4166,6 +4167,7 @@
   bool IsValidVertexGeometry(int vertex_index,ON_TextLog* text_log) const;
   bool IsValidVertexTolerancesAndFlags(int vertex_index,ON_TextLog* text_log) const;
 
+protected:
   void SetTolsFromLegacyValues();
 
   // read helpers to support various versions
