--- opennurbs/opennurbs_object_history.cpp	2024-06-25 15:44:35.950646257 -0400
+++ OPENNURBS_BLD/opennurbs_object_history.cpp	2024-06-25 15:44:06.115134853 -0400
@@ -28,7 +28,7 @@
   // The VALUE_TYPE enum values must never be changed
   // because the values are used to determine the parameter
   // type during file reading.  Additions can be made.
-  enum VALUE_TYPE
+  enum VALUE_TYPE: unsigned int
   {
     no_value_type         =  0,
 
