--- opennurbs/opennurbs_brep.cpp	2024-06-25 15:28:09.117849296 -0400
+++ OPENNURBS_BLD/opennurbs_brep.cpp	2024-06-25 15:26:59.805993657 -0400
@@ -3595,6 +3595,10 @@
     }   
   }
 
+  // BRL-CAD uses this for a tolerance check below
+  const ON_Surface *surf = loop.Face()->SurfaceOf();
+  double urange = (surf) ? (surf->Domain(0)[1] - surf->Domain(0)[0]) : DBL_MAX;
+  double vrange = (surf) ? (surf->Domain(1)[1] - surf->Domain(1)[0]) : DBL_MAX;
 
   // make sure ends of trims jibe
   int ci0, ci1, next_lti;
@@ -3623,6 +3627,12 @@
       //    didn't get flagged as bad.
       double xtol = (fabs(P0.x) + fabs(P1.x))*1.0e-10;
       double ytol = (fabs(P0.y) + fabs(P1.y))*1.0e-10;
+
+      // Oct 12 2009 Rather than using the above check, BRL-CAD uses
+      // relative uv size if it is available
+      xtol = (urange < DBL_MAX) ? urange * trim0.m_tolerance[0] : xtol;
+      ytol = (vrange < DBL_MAX) ? vrange * trim0.m_tolerance[1] : ytol;
+
       if ( xtol < ON_ZERO_TOLERANCE )
         xtol = ON_ZERO_TOLERANCE;
       if ( ytol < ON_ZERO_TOLERANCE )
