--- opennurbs/opennurbs_system_runtime.h	2024-09-24 11:09:52.628564406 -0400
+++ OPENNURBS_BLD/opennurbs_system_runtime.h	2024-09-24 11:18:03.434592812 -0400
@@ -61,6 +61,11 @@
 #define ON_RUNTIME_WASM
 #endif
 
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD__) || defined(__bsdi__)
+#if !defined(ON_RUNTIME_BSD)
+#define ON_RUNTIME_BSD
+#endif
+
 #endif
 /*
 //
@@ -132,7 +137,7 @@
 #define ON_RUNTIME_WIN_WINOS
 #endif
 
-#if defined(_M_X64) || defined(_WIN64)
+#if defined(_M_X64) || defined(_M_ARM64) || defined(_WIN64)
 #define ON_64BIT_RUNTIME
 #elif defined(_M_X86) || defined(_WIN32)
 #define ON_32BIT_RUNTIME
@@ -141,7 +146,7 @@
 #endif
 
 #if !defined(ON_LITTLE_ENDIAN)
-#if (defined(_M_X64) || defined(_M_IX86) || defined (__i386__) || defined( __x86_64__ ))
+#if (defined(_M_X64) || defined(_M_ARM64) || defined(_M_IX86) || defined(__i386__) || defined(__x86_64__) || defined(__x86_64))
 #define ON_LITTLE_ENDIAN
 #endif
 #endif
@@ -151,7 +156,7 @@
 #define ON_RUNTIME_LINUX
 #endif
 
-#if defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM)
+#if defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM) || defined (ON_RUNTIME_BSD)
 
 #if !defined(ON_SIZEOF_WCHAR_T)
 #define ON_SIZEOF_WCHAR_T 4
@@ -164,7 +169,9 @@
 #define ON_32BIT_RUNTIME
 #endif
 
+#if !defined(ON_RUNTIME_BSD)
 #include <endian.h>
+#endif
 #if __BYTE_ORDER == __LITTLE_ENDIAN
 #define ON_LITTLE_ENDIAN
 #else
