--- opennurbs/opennurbs_defines.cpp	2024-09-09 15:56:01.177363986 -0400
+++ OPENNURBS_BLD/opennurbs_defines.cpp	2024-09-09 15:56:35.936771800 -0400
@@ -498,6 +498,9 @@
 #elif defined(ON_RUNTIME_APPLE) || defined(ON_RUNTIME_ANDROID)
   //fcloseall is not supported on mac/ios or android
   return EOF;
+#elif defined(BSD) || defined(__FreeBSD__)
+  fcloseall();
+  return 0;
 #else
   return fcloseall();
 #endif
