--- opennurbs/example_brep/example_brep.cpp	2024-06-25 15:44:35.959646109 -0400
+++ OPENNURBS_BLD/example_brep/example_brep.cpp	2024-06-25 15:44:06.087135312 -0400
@@ -589,9 +589,9 @@
     brep = NULL;
   }
 
-  //ON_BOOL32 bIsManifold;
-  //ON_BOOL32 bHasBoundary;
-  //ON_BOOL32 b = brep->IsManifold( &bIsManifold,&bHasBoundary );
+  //bool bIsManifold;
+  //bool bHasBoundary;
+  //bool b = brep->IsManifold( &bIsManifold,&bHasBoundary );
 
   return brep;
 }
