--- opennurbs/opennurbs_lookup.cpp	2023-12-21 11:57:27.113240837 -0500
+++ OPENNURBS_BLD/opennurbs_lookup.cpp	2023-12-21 11:58:28.000088916 -0500
@@ -696,6 +696,9 @@
   return m_active_id_count;
 }
 
+// NOTE - optimization currently disabled for FirstElement due to compiler bug
+// in VS2022 - see https://github.com/microsoft/vcpkg/issues/19561
+#pragma optimize("", off)
 ON_SerialNumberMap::SN_ELEMENT* ON_SerialNumberMap::FirstElement() const
 {
   SN_ELEMENT* e=nullptr;
@@ -756,6 +759,7 @@
   }
   return e;
 }
+#pragma optimize("", on)
 
 struct ON_SerialNumberMap::SN_ELEMENT* ON_SerialNumberMap::LastElement() const
 {
