--- opennurbs/opennurbs_locale.cpp	2024-02-16 13:44:43.184776025 -0500
+++ OPENNURBS_BLD/opennurbs_locale.cpp	2024-02-16 14:21:56.703145243 -0500
@@ -32,7 +32,7 @@
     ON_C_locale = _create_locale(LC_ALL, "C");
 #elif defined(ON_RUNTIME_APPLE)
     ON_C_locale = _c_locale;
-#elif defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#elif defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
     ON_C_locale = 0;
 #else
     ON_C_locale = _create_locale(category, locale);
@@ -101,7 +101,7 @@
     }
   }
   return newlocale(LC_ALL_MASK, apple_name, ON_CRT_C_locale() );
-#elif defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#elif defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   return 0;
 #else
   return _create_locale(category, locale);
@@ -1341,7 +1341,7 @@
   static bool Validate_sprintf_l()
   {
 #if defined(ON_COMPILER_CLANG) || defined(ON_COMPILER_GNU)
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
     // Test formatted printing
     char buffer[64] = { 0 };
     // Testing C-runtime - do not using ON_String::FormatIntoBuffer
@@ -1366,7 +1366,7 @@
   static bool Validate_sprintf_s_l()
   {
 #if defined(ON_COMPILER_CLANG) || defined(ON_COMPILER_GNU)
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
     // Test formatted printing
     char buffer[64] = { 0 };
     size_t buffer_capacity = (sizeof(buffer) / sizeof(buffer[0])) - 1;
@@ -1420,7 +1420,7 @@
   static bool Validate_sscanf_l()
   {
 #if defined(ON_COMPILER_CLANG) || defined(ON_COMPILER_GNU)
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
     // Test formatted scanning
     double a = ON_UNSET_VALUE;
     // Testing C-runtime - do not using ON_String::Scan
@@ -1445,7 +1445,7 @@
   static bool Validate_sscanf_s_l()
   {
 #if defined(ON_COMPILER_CLANG) || defined(ON_COMPILER_GNU)
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
     // Test formatted scanning
     double a = ON_UNSET_VALUE;
     // Testing C-runtime - do not using ON_String::Scan
