--- opennurbs/opennurbs_font.cpp	2023-12-20 11:23:16.153987006 -0500
+++ OPENNURBS_BLD/opennurbs_font.cpp	2023-12-20 11:22:15.347051556 -0500
@@ -8936,7 +8936,7 @@
     const wchar_t* prev_clean_family_name = nullptr;
     for (size_t i = 0; i < installed_count; i++)
     {
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
       const ON_Font* font = installed_fonts[(int)i];
 #else
       const ON_Font* font = installed_fonts[i];
@@ -8999,7 +8999,7 @@
       InternalHashToName candidate;
       for (size_t i = 0; i < count0; i++)
       {
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
         InternalHashToName e = a[(int)i];
 #else
         InternalHashToName e = a[i];
