--- opennurbs/opennurbs_string_format.cpp	2024-06-25 15:44:35.953646208 -0400
+++ OPENNURBS_BLD/opennurbs_string_format.cpp	2024-06-25 15:44:06.131134592 -0400
@@ -802,7 +802,7 @@
   // CLang modifies args so a copy is required
   va_list args_copy;
   va_copy (args_copy, args);
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   int len = vsnprintf(buffer, buffer_capacity, format, args_copy);
 #else
   int len = vsnprintf_l(buffer, buffer_capacity, ON_Locale::Ordinal.NumericLocalePtr(), format, args_copy);
@@ -855,7 +855,7 @@
   va_list args_copy;
   va_copy (args_copy, args);
 
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   int len = vsnprintf(nullptr, 0, format, args_copy);
 #else
   int len = vsnprintf_l(nullptr, 0, ON_Locale::Ordinal.NumericLocalePtr(), format, args_copy);
