--- opennurbs/opennurbs_brep.h	2024-06-25 15:28:09.117849296 -0400
+++ OPENNURBS_BLD/opennurbs_brep.h	2024-06-25 15:26:59.814993508 -0400
@@ -4152,6 +4152,7 @@
         );
 
   // helpers for validation checking
+public:
   bool IsValidTrim(int trim_index,ON_TextLog* text_log) const;
   bool IsValidTrimTopology(int trim_index,ON_TextLog* text_log) const;
   bool IsValidTrimGeometry(int trim_index,ON_TextLog* text_log) const;
@@ -4177,6 +4178,7 @@
   bool IsValidVertexGeometry(int vertex_index,ON_TextLog* text_log) const;
   bool IsValidVertexTolerancesAndFlags(int vertex_index,ON_TextLog* text_log) const;
 
+protected:
   void SetTolsFromLegacyValues();
 
   // read helpers to support various versions
