--- opennurbs/opennurbs_system.h	2024-02-16 13:44:43.192775892 -0500
+++ OPENNURBS_BLD/opennurbs_system.h	2024-02-16 14:38:32.266292178 -0500
@@ -494,7 +494,7 @@
 #pragma ON_PRAGMA_WARNING_AFTER_DIRTY_INCLUDE
 
 #pragma ON_PRAGMA_WARNING_BEFORE_DIRTY_INCLUDE
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
 #include "android_uuid/uuid.h"
 #else
 #include <uuid/uuid.h>
