--- opennurbs/opennurbs_optimize.h	2024-02-16 13:44:43.188775957 -0500
+++ OPENNURBS_BLD/opennurbs_optimize.h	2024-02-16 14:27:19.549703573 -0500
@@ -15,6 +15,7 @@
 #define OPENNURBS_OPTIMIZE_INC_
 
 // find a local minimum of a 1 parameter function
+ON_DECL
 int ON_FindLocalMinimum( // returns 0 - failed to converge, 1 - success, 2 - failed to converge to requested tolerances
         int (*)(void*,double,double*,double*), // f(void*, double t, double* value, double* derivative );
         void*, // passed as the void* argument to the above function
