--- opennurbs/opennurbs_uuid.cpp	2024-06-25 15:44:35.957646142 -0400
+++ OPENNURBS_BLD/opennurbs_uuid.cpp	2024-06-25 15:44:06.142134412 -0400
@@ -12,7 +12,7 @@
 ////////////////////////////////////////////////////////////////
 
 #include "opennurbs.h"
-#if defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
 #include "android_uuid/uuid.h"
 #endif
 #if !defined(ON_COMPILING_OPENNURBS)
@@ -221,7 +221,7 @@
   return true;
 #else
 
-#if defined(ON_RUNTIME_LINUX)
+#if defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_BSD)
   uuid_generate((unsigned char*)&new_uuid);
   return true;
 #else
