--- opennurbs/opennurbs_font.cpp	2024-09-24 10:49:49.815389242 -0400
+++ OPENNURBS_BLD/opennurbs_font.cpp	2024-09-24 10:51:07.967027619 -0400
@@ -8936,7 +8936,7 @@
     const wchar_t* prev_clean_family_name = nullptr;
     for (size_t i = 0; i < installed_count; i++)
     {
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM) || defined(ON_RUNTIME_BSD)
       const ON_Font* font = installed_fonts[(int)i];
 #else
       const ON_Font* font = installed_fonts[i];
@@ -8999,7 +8999,7 @@
       InternalHashToName candidate;
       for (size_t i = 0; i < count0; i++)
       {
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM) || defined(ON_RUNTIME_BSD)
         InternalHashToName e = a[(int)i];
 #else
         InternalHashToName e = a[i];
