--- opennurbs/opennurbs_polyline.cpp	2024-09-24 10:54:54.798089595 -0400
+++ OPENNURBS_BLD/opennurbs_polyline.cpp	2024-09-24 10:55:48.500159662 -0400
@@ -455,7 +455,7 @@
   maxN = maxN.UnitVector();
   for (size_t i = 0; i < point_count; ++i)
   {
-#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM)
+#if defined(ON_RUNTIME_ANDROID) || defined(ON_RUNTIME_LINUX) || defined(ON_RUNTIME_WASM) || defined(ON_RUNTIME_BSD)
     double d = maxN * C[(unsigned int)i];
 #else
     double d = maxN * C[i];
