diff -Naur OpenColorIO/CMakeLists.txt OPENCOLORIO_BLD/CMakeLists.txt
--- OpenColorIO/CMakeLists.txt	2023-12-20 15:33:40.548452208 -0500
+++ OPENCOLORIO_BLD/CMakeLists.txt	2023-12-20 15:33:17.076776842 -0500
@@ -436,7 +436,7 @@
    application or shared library that uses static OpenColorIO.
 
    The following mandatory dependencies MUST be linked to the consumer application or shared library:
-   expat, yaml-cpp, Imath, pystring, minizip-ng and ZLIB")
+   expat, yaml-cpp, Imath, pystring, minizip and ZLIB")
 endif()
 
 
@@ -523,8 +523,6 @@
         ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/Findexpat.cmake
         ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/FindImath.cmake
         ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/Findpystring.cmake
-        ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/Findminizip-ng.cmake
-        ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/Findyaml-cpp.cmake
         DESTINATION ${OCIO_CUSTOM_FIND_MODULE_DIR}
     )
 endif()
diff -Naur OpenColorIO/share/cmake/modules/FindExtPackages.cmake OPENCOLORIO_BLD/share/cmake/modules/FindExtPackages.cmake
--- OpenColorIO/share/cmake/modules/FindExtPackages.cmake	2023-12-20 15:33:40.572451873 -0500
+++ OPENCOLORIO_BLD/share/cmake/modules/FindExtPackages.cmake	2023-12-20 15:33:17.076776842 -0500
@@ -75,7 +75,7 @@
 
 # Imath (>=3.1)
 # https://github.com/AcademySoftwareFoundation/Imath
-ocio_handle_dependency(  Imath REQUIRED ALLOW_INSTALL
+ocio_handle_dependency(  OpenEXR REQUIRED ALLOW_INSTALL
                          MIN_VERSION 3.1.1
                          RECOMMENDED_VERSION 3.1.6
                          RECOMMENDED_VERSION_REASON "Latest version tested with OCIO")
@@ -110,10 +110,11 @@
 
 # minizip-ng
 # https://github.com/zlib-ng/minizip-ng
-ocio_handle_dependency(  minizip-ng REQUIRED ALLOW_INSTALL
-                         MIN_VERSION 3.0.6
-                         RECOMMENDED_VERSION 3.0.7
-                         RECOMMENDED_VERSION_REASON "Latest version tested with OCIO")
+find_package(minizip REQUIRED)
+#ocio_handle_dependency(  minizip REQUIRED ALLOW_INSTALL
+#                         MIN_VERSION 3.0.6
+#                         RECOMMENDED_VERSION 3.0.7
+#                         RECOMMENDED_VERSION_REASON "Latest version tested with OCIO")
 
 ###############################################################################
 ##
diff -Naur OpenColorIO/share/cmake/modules/Findpystring.cmake OPENCOLORIO_BLD/share/cmake/modules/Findpystring.cmake
--- OpenColorIO/share/cmake/modules/Findpystring.cmake	2023-12-20 15:33:40.572451873 -0500
+++ OPENCOLORIO_BLD/share/cmake/modules/Findpystring.cmake	2023-12-20 15:33:17.080776786 -0500
@@ -42,7 +42,7 @@
         NAMES
             pystring libpystring
         HINTS
-            ${_pystring_SEARCH_DIRS}
+            ${pystring_ROOT}
         PATH_SUFFIXES
             pystring/lib
             lib64
@@ -81,4 +81,4 @@
     )
 
     mark_as_advanced(pystring_INCLUDE_DIR pystring_LIBRARY pystring_VERSION)
-endif()
\ No newline at end of file
+endif()
diff -Naur OpenColorIO/src/apps/ocioarchive/CMakeLists.txt OPENCOLORIO_BLD/src/apps/ocioarchive/CMakeLists.txt
--- OpenColorIO/src/apps/ocioarchive/CMakeLists.txt	2023-12-20 15:33:40.584451706 -0500
+++ OPENCOLORIO_BLD/src/apps/ocioarchive/CMakeLists.txt	2023-12-20 15:33:17.080776786 -0500
@@ -21,7 +21,7 @@
     PRIVATE
         apputils
         OpenColorIO
-        MINIZIP::minizip-ng
+        MINIZIP::minizip
 )
 
 include(StripUtils)
diff -Naur OpenColorIO/src/cmake/Config.cmake.in OPENCOLORIO_BLD/src/cmake/Config.cmake.in
--- OpenColorIO/src/cmake/Config.cmake.in	2023-12-20 15:33:40.592451595 -0500
+++ OPENCOLORIO_BLD/src/cmake/Config.cmake.in	2023-12-20 15:33:17.080776786 -0500
@@ -53,8 +53,8 @@
         endif()
     endif()
 
-    if (NOT TARGET MINIZIP::minizip-ng)
-        find_dependency(minizip-ng @minizip-ng_VERSION@)
+    if (NOT TARGET MINIZIP::minizip)
+        find_dependency(minizip @minizip_VERSION@)
     endif()
 
     # Remove OCIO custom find module path.
diff -Naur OpenColorIO/src/OpenColorIO/CMakeLists.txt OPENCOLORIO_BLD/src/OpenColorIO/CMakeLists.txt
--- OpenColorIO/src/OpenColorIO/CMakeLists.txt	2023-12-20 15:33:40.576451819 -0500
+++ OPENCOLORIO_BLD/src/OpenColorIO/CMakeLists.txt	2023-12-20 15:37:00.877582518 -0500
@@ -309,8 +309,8 @@
         "$<BUILD_INTERFACE:utils::from_chars>"
         "$<BUILD_INTERFACE:utils::strings>"
         "$<BUILD_INTERFACE:xxHash>"
-        ${YAML_CPP_LIBRARIES}
-        MINIZIP::minizip-ng
+        yaml-cpp::yaml-cpp
+        MINIZIP::minizip
 )
 
 if(OCIO_USE_SIMD AND OCIO_USE_SSE2NEON AND COMPILER_SUPPORTS_SSE_WITH_SSE2NEON)
@@ -388,20 +388,20 @@
     endif()
 
 
-    # Check for minizip first since our Findminizip module sets minizip-ng_LIBRARY.
-    if (minizip_LIBRARY OR minizip-ng_COMPAT)
-        get_filename_component(_minizip-ng_LIBDIR "${minizip_LIBRARY}" DIRECTORY)
-        set(_minizip-ng_NAME "minizip")
-    elseif(minizip-ng_LIBRARY)
-        get_filename_component(_minizip-ng_LIBDIR "${minizip-ng_LIBRARY}" DIRECTORY)
-        set(_minizip-ng_NAME "minizip-ng")
+    # Check for minizip first since our Findminizip module sets minizip_LIBRARY.
+    if (minizip_LIBRARY OR minizip_COMPAT)
+        get_filename_component(_minizip_LIBDIR "${minizip_LIBRARY}" DIRECTORY)
+        set(_minizip_NAME "minizip")
+    elseif(minizip_LIBRARY)
+        get_filename_component(_minizip_LIBDIR "${minizip_LIBRARY}" DIRECTORY)
+        set(_minizip_NAME "minizip")
     endif()
 
-    if (_minizip-ng_LIBDIR)
+    if (_minizip_LIBDIR)
         # Add the path to CMake list of search paths for libraries.
-        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-L${_minizip-ng_LIBDIR}")
-        # Hide the minizip-ng symbols.
-        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-hidden-l${_minizip-ng_NAME}")
+        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-L${_minizip_LIBDIR}")
+        # Hide the minizip symbols.
+        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-hidden-l${_minizip_NAME}")
     endif()
 
     if (_OCIO_LINK_FLAGS_LIST_)
diff -Naur OpenColorIO/tests/cpu/CMakeLists.txt OPENCOLORIO_BLD/tests/cpu/CMakeLists.txt
--- OpenColorIO/tests/cpu/CMakeLists.txt	2023-12-20 15:33:40.592451595 -0500
+++ OPENCOLORIO_BLD/tests/cpu/CMakeLists.txt	2023-12-20 15:33:17.080776786 -0500
@@ -37,7 +37,7 @@
             utils::strings
             ${YAML_CPP_LIBRARIES}
             testutils
-            MINIZIP::minizip-ng
+            MINIZIP::minizip
             xxHash
     )
 
