diff -Naur OpenColorIO/CMakeLists.txt OPENCOLORIO_BLD/CMakeLists.txt
--- OpenColorIO/CMakeLists.txt	2023-12-20 15:33:40.548452208 -0500
+++ OPENCOLORIO_BLD/CMakeLists.txt	2023-12-20 15:33:17.076776842 -0500
@@ -436,7 +436,7 @@
    application or shared library that uses static OpenColorIO.
 
    The following mandatory dependencies MUST be linked to the consumer application or shared library:
-   expat, yaml-cpp, Imath, pystring, minizip-ng and ZLIB")
+   expat, yaml-cpp, Imath, pystring, minizip and ZLIB")
 endif()
 
 
@@ -523,8 +523,6 @@
         ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/Findexpat.cmake
         ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/FindImath.cmake
         ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/Findpystring.cmake
-        ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/Findminizip-ng.cmake
-        ${CMAKE_CURRENT_LIST_DIR}/share/cmake/modules/Findyaml-cpp.cmake
         DESTINATION ${OCIO_CUSTOM_FIND_MODULE_DIR}
     )
 endif()
diff -Naur OpenColorIO/share/cmake/modules/FindExtPackages.cmake OPENCOLORIO_BLD/share/cmake/modules/FindExtPackages.cmake
--- OpenColorIO/share/cmake/modules/FindExtPackages.cmake	2023-12-20 15:33:40.572451873 -0500
+++ OPENCOLORIO_BLD/share/cmake/modules/FindExtPackages.cmake	2023-12-20 15:33:17.076776842 -0500
@@ -75,7 +75,7 @@
 
 # Imath (>=3.1)
 # https://github.com/AcademySoftwareFoundation/Imath
-ocio_handle_dependency(  Imath REQUIRED ALLOW_INSTALL
+ocio_handle_dependency(  OpenEXR REQUIRED ALLOW_INSTALL
                          MIN_VERSION 3.1.1
                          RECOMMENDED_VERSION 3.1.6
                          RECOMMENDED_VERSION_REASON "Latest version tested with OCIO")
@@ -110,10 +110,11 @@
 
 # minizip-ng
 # https://github.com/zlib-ng/minizip-ng
-ocio_handle_dependency(  minizip-ng REQUIRED ALLOW_INSTALL
-                         MIN_VERSION 3.0.6
-                         RECOMMENDED_VERSION 3.0.7
-                         RECOMMENDED_VERSION_REASON "Latest version tested with OCIO")
+find_package(minizip REQUIRED)
+#ocio_handle_dependency(  minizip REQUIRED ALLOW_INSTALL
+#                         MIN_VERSION 3.0.6
+#                         RECOMMENDED_VERSION 3.0.7
+#                         RECOMMENDED_VERSION_REASON "Latest version tested with OCIO")
 
 ###############################################################################
 ##
diff -Naur OpenColorIO/share/cmake/modules/Findminizip-ng.cmake OPENCOLORIO_BLD/share/cmake/modules/Findminizip-ng.cmake
--- OpenColorIO/share/cmake/modules/Findminizip-ng.cmake	2023-12-20 15:33:40.572451873 -0500
+++ OPENCOLORIO_BLD/share/cmake/modules/Findminizip-ng.cmake	1969-12-31 19:00:00.000000000 -0500
@@ -1,220 +0,0 @@
-# SPDX-License-Identifier: BSD-3-Clause
-# Copyright Contributors to the OpenColorIO Project.
-#
-# Locate minizip-ng
-#
-# Variables defined by this module:
-#   minizip-ng_FOUND        - If FALSE, do not try to link to minizip-ng
-#   minizip-ng_LIBRARY      - minizip-ng library to link to
-#   minizip-ng_INCLUDE_DIR  - Where to find mz.h and other headers
-#   minizip-ng_VERSION      - The version of the library
-#   minizip-ng_COMPAT       - Whether minizip-ng MZ_COMPAT was used or not
-#
-# Global targets defined by this module:
-#   MINIZIP::minizip-ng - IMPORTED target, if found
-#
-# Usually CMake will use the dynamic library rather than static, if both are present. 
-# In this case, you may set minizip-ng_STATIC_LIBRARY to ON to request use of the static one. 
-# If only the static library is present (such as when OCIO builds the dependency), then the option 
-# is not needed.
-#
-# If the library is not installed in a typical location where CMake will find it, you may specify 
-# the location using one of the following methods:
-# -- Set -Dminizip-ng_DIR to point to the directory containing the CMake configuration file for the package.
-# -- Set -Dminizip-ng_ROOT to point to the directory containing the lib and include directories.
-# -- Set -Dminizip-ng_LIBRARY and -Dminizip-ng_INCLUDE_DIR to point to the lib and include directories.
-#
-#
-# For external builds of minizip-ng, please note that the same build options should be used.
-# Using more options, such as enabling other compression methods, will provoke linking issue
-# since OCIO is not linking to those libraries.
-#
-# e.g. Setting MZ_BZIP2=ON will cause linking issue since OCIO will not be linked against BZIP2.
-#
-
-###############################################################################
-### Try to find package ###
-
-# BUILD_TYPE_DEBUG variable is currently set in one of the OCIO's CMake files. 
-# Now that some OCIO's find module are installed with the library itself (with static build), 
-# a consumer app don't have access to the variables set by an OCIO's CMake files. Therefore, some 
-# OCIO's find modules must detect the build type by itselves. 
-set(BUILD_TYPE_DEBUG OFF)
-if(CMAKE_BUILD_TYPE MATCHES "[Dd][Ee][Bb][Uu][Gg]")
-   set(BUILD_TYPE_DEBUG ON)
-endif()
-
-if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
-    if(NOT DEFINED minizip-ng_ROOT)
-        # Search for minizip-ng-config.cmake
-        find_package(minizip-ng ${minizip-ng_FIND_VERSION} CONFIG QUIET)
-    endif()
-    
-    if (minizip-ng_FOUND)
-        get_target_property(minizip-ng_INCLUDE_DIR MINIZIP::minizip-ng INTERFACE_INCLUDE_DIRECTORIES)
-        get_target_property(minizip-ng_LIBRARY MINIZIP::minizip-ng LOCATION)
-        
-        # Depending on the options used when minizip-ng was built, it could have multiple libraries
-        # listed in INTERFACE_LINK_LIBRARIES. OCIO only needs ZLIB.
-        # Only add custom zlib target ZLIB::ZLIB to INTERFACE_LINK_LIBRARIES.
-        set_target_properties(MINIZIP::minizip-ng PROPERTIES INTERFACE_LINK_LIBRARIES "ZLIB::ZLIB")
-
-        if (NOT minizip-ng_LIBRARY)
-            # Lib names to search for
-            set(_minizip-ng_LIB_NAMES minizip-ng)
-
-            if(BUILD_TYPE_DEBUG)
-                # Prefer Debug lib names (Windows only)
-                list(INSERT _minizip-ng_LIB_NAMES 0 minizip-ngd)
-            endif()
-
-            if(minizip-ng_STATIC_LIBRARY)
-                # Prefer static lib names
-                set(_minizip-ng_STATIC_LIB_NAMES 
-                    "${CMAKE_STATIC_LIBRARY_PREFIX}minizip-ng${CMAKE_STATIC_LIBRARY_SUFFIX}")
-                if(WIN32 AND BUILD_TYPE_DEBUG)
-                    # Prefer static Debug lib names (Windows only)
-                    list(INSERT _minizip-ng_STATIC_LIB_NAMES 0
-                        "${CMAKE_STATIC_LIBRARY_PREFIX}minizip-ngd${CMAKE_STATIC_LIBRARY_SUFFIX}")
-                endif()
-            endif()
-
-            # Find library
-            find_library(minizip-ng_LIBRARY
-                NAMES
-                    ${_minizip-ng_STATIC_LIB_NAMES}
-                    ${_minizip-ng_LIB_NAMES}
-                HINTS
-                    ${minizip-ng_ROOT}
-                    ${PC_minizip-ng_LIBRARY_DIRS}
-                PATH_SUFFIXES
-                    lib64 lib 
-            )
-
-            # Set IMPORTED_LOCATION property for MINIZIP::minizip-ng target.
-            if (TARGET MINIZIP::minizip-ng)
-                set_target_properties(MINIZIP::minizip-ng PROPERTIES
-                    IMPORTED_LOCATION "${minizip-ng_LIBRARY}"
-                )
-            endif()
-        endif()
-    else()
-        list(APPEND _minizip-ng_REQUIRED_VARS minizip-ng_INCLUDE_DIR)
-
-        # Search for minizip-ng.pc
-        find_package(PkgConfig QUIET)
-        pkg_check_modules(PC_minizip-ng QUIET "minizip-ng>=${minizip-ng_FIND_VERSION}")
-
-        # Find include directory
-        find_path(minizip-ng_INCLUDE_DIR
-            NAMES
-                mz.h
-            HINTS
-                ${minizip-ng_ROOT}
-                ${PC_minizip-ng_INCLUDE_DIRS}
-            PATH_SUFFIXES
-                include
-                include/minizip-ng
-                include/minizip
-                minizip-ng/include
-                minizip/include
-        )
-
-        # Minizip-ng uses prefix "lib" on all platform by default.
-        # Library name doesn't change in debug.
-
-        # Lib names to search for.
-        # Search for "minizip" since the library could be named "minizip" if it was built 
-        # with MZ_COMPAT=ON.
-        set(_minizip-ng_LIB_NAMES libminizip-ng minizip-ng libminizip minizip)
-
-        if(minizip-ng_STATIC_LIBRARY)
-            # Looking for both "lib" prefix and CMAKE_STATIC_LIBRARY_PREFIX.
-            # Prefer static lib names.
-            set(_minizip-ng_STATIC_LIB_NAMES 
-                "libminizip-ng${CMAKE_STATIC_LIBRARY_SUFFIX}"
-                "${CMAKE_STATIC_LIBRARY_PREFIX}minizip-ng${CMAKE_STATIC_LIBRARY_SUFFIX}"
-                "libminizip${CMAKE_STATIC_LIBRARY_SUFFIX}"
-                "${CMAKE_STATIC_LIBRARY_PREFIX}minizip${CMAKE_STATIC_LIBRARY_SUFFIX}")
-        endif()
-
-        # Find library
-        find_library(minizip-ng_LIBRARY
-            NAMES
-                ${_minizip-ng_STATIC_LIB_NAMES}
-                ${_minizip-ng_LIB_NAMES}
-            HINTS
-                ${minizip-ng_ROOT}
-                ${PC_minizip-ng_LIBRARY_DIRS}
-            PATH_SUFFIXES
-                lib64 lib 
-        )
-
-        # Get version from header or pkg-config
-        if(minizip-ng_INCLUDE_DIR)
-            list(GET minizip-ng_INCLUDE_DIR 0 _minizip-ng_INCLUDE_DIR)
-            if(EXISTS "${_minizip-ng_INCLUDE_DIR}/mz.h")
-                set(_minizip-ng_CONFIG "${_minizip-ng_INCLUDE_DIR}/mz.h")
-            endif()
-        endif()
-
-        if(_minizip-ng_CONFIG)
-            file(STRINGS "${_minizip-ng_CONFIG}" _minizip-ng_VER_SEARCH 
-                REGEX "^[ \t]*#define[ \t]+MZ_VERSION[ \t]+\\(\"[.0-9]+\"\\).*$")
-            if(_minizip-ng_VER_SEARCH)
-                string(REGEX REPLACE ".*#define[ \t]+MZ_VERSION[ \t]+\\(\"([.0-9]+)\"\\).*" 
-                    "\\1" minizip-ng_VERSION "${_minizip-ng_VER_SEARCH}")
-            endif()
-        elseif(PC_minizip-ng_FOUND)
-            set(minizip-ng_VERSION "${PC_minizip-ng_VERSION}")
-        endif()
-    endif()
-
-    # Override REQUIRED if package can be installed
-    if(OCIO_INSTALL_EXT_PACKAGES STREQUAL MISSING)
-        set(minizip-ng_FIND_REQUIRED FALSE)
-    endif()
-
-    include(FindPackageHandleStandardArgs)
-    find_package_handle_standard_args(minizip-ng
-        REQUIRED_VARS 
-            minizip-ng_LIBRARY
-            minizip-ng_INCLUDE_DIR
-        VERSION_VAR
-            minizip-ng_VERSION
-    )
-endif()
-
-###############################################################################
-### Create target
-
-if(minizip-ng_FOUND AND NOT TARGET MINIZIP::minizip-ng)
-    add_library(MINIZIP::minizip-ng UNKNOWN IMPORTED GLOBAL)
-    set(_minizip-ng_TARGET_CREATE TRUE)
-endif()
-
-###############################################################################
-### Configure target ###
-
-if(_minizip-ng_TARGET_CREATE)
-    set_target_properties(MINIZIP::minizip-ng PROPERTIES
-        IMPORTED_LOCATION "${minizip-ng_LIBRARY}"
-        INTERFACE_INCLUDE_DIRECTORIES "${minizip-ng_INCLUDE_DIR}"
-    )
-
-    mark_as_advanced(minizip-ng_INCLUDE_DIR minizip-ng_LIBRARY minizip-ng_VERSION)
-
-    target_link_libraries(MINIZIP::minizip-ng INTERFACE ZLIB::ZLIB)
-endif()
-
-###############################################################################
-### Detect compatibility mode ###
-
-set(minizip-ng_COMPAT FALSE)
-if(minizip-ng_INCLUDE_DIR)
-    list(GET minizip-ng_INCLUDE_DIR 0 _minizip-ng_INCLUDE_DIR)
-    if(EXISTS "${_minizip-ng_INCLUDE_DIR}/mz_compat.h")
-        set(minizip-ng_COMPAT TRUE)
-    endif()
-endif()
-mark_as_advanced(minizip-ng_COMPAT)
diff -Naur OpenColorIO/share/cmake/modules/Findpystring.cmake OPENCOLORIO_BLD/share/cmake/modules/Findpystring.cmake
--- OpenColorIO/share/cmake/modules/Findpystring.cmake	2023-12-20 15:33:40.572451873 -0500
+++ OPENCOLORIO_BLD/share/cmake/modules/Findpystring.cmake	2023-12-20 15:33:17.080776786 -0500
@@ -42,7 +42,7 @@
         NAMES
             pystring libpystring
         HINTS
-            ${_pystring_SEARCH_DIRS}
+            ${pystring_ROOT}
         PATH_SUFFIXES
             pystring/lib
             lib64
@@ -81,4 +81,4 @@
     )
 
     mark_as_advanced(pystring_INCLUDE_DIR pystring_LIBRARY pystring_VERSION)
-endif()
\ No newline at end of file
+endif()
diff -Naur OpenColorIO/share/cmake/modules/Findyaml-cpp.cmake OPENCOLORIO_BLD/share/cmake/modules/Findyaml-cpp.cmake
--- OpenColorIO/share/cmake/modules/Findyaml-cpp.cmake	2023-12-20 15:33:40.572451873 -0500
+++ OPENCOLORIO_BLD/share/cmake/modules/Findyaml-cpp.cmake	1969-12-31 19:00:00.000000000 -0500
@@ -1,172 +0,0 @@
-# SPDX-License-Identifier: BSD-3-Clause
-# Copyright Contributors to the OpenColorIO Project.
-#
-# Locate yaml-cpp
-#
-# Variables defined by this module:
-#   yaml-cpp_FOUND          - Indicate whether the library was found or not
-#   yaml-cpp_LIBRARY        - Path to the library file
-#   yaml-cpp_INCLUDE_DIR    - Location of the header files
-#   yaml-cpp_VERSION        - Library's version
-#
-# Global targets defined by this module:
-#   yaml-cpp::yaml-cpp
-#
-# For compatibility with the upstream CMake package, the following variables and targets are defined:
-#   YAML_CPP_LIBRARIES      - Libraries to link against yaml-cpp
-#   YAML_CPP_INCLUDE_DIR    - Include directory
-#
-# Usually CMake will use the dynamic library rather than static, if both are present. 
-# In this case, you may set yaml-cpp_STATIC_LIBRARY to ON to request use of the static one. 
-# If only the static library is present (such as when OCIO builds the dependency), then the option 
-# is not needed.
-#
-# If the library is not installed in a typical location where CMake will find it, you may specify 
-# the location using one of the following methods:
-# -- Set -Dyaml-cpp_DIR to point to the directory containing the CMake configuration file for the package.
-# -- Set -Dyaml-cpp_ROOT to point to the directory containing the lib and include directories.
-# -- Set -Dyaml-cpp_LIBRARY and -Dyaml-cpp_INCLUDE_DIR to point to the lib and include directories.
-#
-
-###############################################################################
-### Try to find package ###
-
-# BUILD_TYPE_DEBUG variable is currently set in one of the OCIO's CMake files. 
-# Now that some OCIO's find module are installed with the library itself (with static build), 
-# a consumer app don't have access to the variables set by an OCIO's CMake files. Therefore, some 
-# OCIO's find modules must detect the build type by itselves.  
-set(BUILD_TYPE_DEBUG OFF)
-if(CMAKE_BUILD_TYPE MATCHES "[Dd][Ee][Bb][Uu][Gg]")
-   set(BUILD_TYPE_DEBUG ON)
-endif()
-
-if(yaml-cpp_FIND_QUIETLY)
-    set(quiet QUIET)
-endif()
-
-if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
-    set(_yaml-cpp_REQUIRED_VARS yaml-cpp_LIBRARY)
-
-    # Search for yaml-cpp-config.cmake
-    if(NOT DEFINED yaml-cpp_ROOT)
-        find_package(yaml-cpp ${yaml-cpp_FIND_VERSION} CONFIG ${quiet})
-    endif()
-
-    if(yaml-cpp_FOUND)
-        # Alias target for yaml-cpp < 0.8 compatibility
-        if(TARGET yaml-cpp AND NOT TARGET yaml-cpp::yaml-cpp)
-            add_library(yaml-cpp::yaml-cpp ALIAS yaml-cpp)
-        endif()
-
-        get_target_property(yaml-cpp_INCLUDE_DIR yaml-cpp::yaml-cpp INTERFACE_INCLUDE_DIRECTORIES)
-        get_target_property(yaml-cpp_LIBRARY yaml-cpp::yaml-cpp LOCATION)
-    else()
-
-        # As yaml-cpp-config.cmake search fails, search an installed library
-        # using yaml-cpp.pc .
-
-        list(APPEND _yaml-cpp_REQUIRED_VARS yaml-cpp_INCLUDE_DIR yaml-cpp_VERSION)
-
-        # Search for yaml-cpp.pc
-        find_package(PkgConfig ${quiet})
-        pkg_check_modules(PC_yaml-cpp ${quiet} "yaml-cpp>=${yaml-cpp_FIND_VERSION}")
-
-        # Try to detect the version installed, if any.
-        if(NOT PC_yaml-cpp_FOUND)
-            pkg_search_module(PC_yaml-cpp ${quiet} "yaml-cpp")
-        endif()
-
-        # Find include directory
-        find_path(yaml-cpp_INCLUDE_DIR 
-            NAMES
-                yaml-cpp/yaml.h
-            HINTS
-                ${yaml-cpp_ROOT}
-                ${PC_yaml-cpp_INCLUDE_DIRS}
-            PATH_SUFFIXES 
-                include
-                yaml-cpp/include
-                YAML_CPP/include
-        )
-
-        # Lib names to search for
-        set(_yaml-cpp_LIB_NAMES yaml-cpp)
-        if(BUILD_TYPE_DEBUG)
-            # Prefer Debug lib names.
-            list(INSERT _yaml-cpp_LIB_NAMES 0 yaml-cppd)
-        endif()
-
-        if(yaml-cpp_STATIC_LIBRARY)
-            # Prefer static lib names
-            set(_yaml-cpp_STATIC_LIB_NAMES 
-                "${CMAKE_STATIC_LIBRARY_PREFIX}yaml-cpp${CMAKE_STATIC_LIBRARY_SUFFIX}")
-
-            # Starting from 0.7.0, all platforms uses the suffix "d" for debug.
-            # See https://github.com/jbeder/yaml-cpp/blob/master/CMakeLists.txt#L141
-            if(BUILD_TYPE_DEBUG)
-                list(INSERT _yaml-cpp_STATIC_LIB_NAMES 0
-                    "${CMAKE_STATIC_LIBRARY_PREFIX}yaml-cppd${CMAKE_STATIC_LIBRARY_SUFFIX}")
-            endif()
-        endif()
-
-        # Find library
-        find_library(yaml-cpp_LIBRARY
-            NAMES 
-                ${_yaml-cpp_STATIC_LIB_NAMES}
-                ${_yaml-cpp_LIB_NAMES}
-            HINTS 
-                ${yaml-cpp_ROOT}
-                ${PC_yaml-cpp_LIBRARY_DIRS}
-            PATH_SUFFIXES 
-                lib64
-                lib
-                YAML_CPP/lib
-        )
-
-        # Get version from pkg-config if it was found.
-        if(PC_yaml-cpp_FOUND)
-            set(yaml-cpp_VERSION "${PC_yaml-cpp_VERSION}")
-        endif()
-    endif()
-
-    # Override REQUIRED if package can be installed
-    if(OCIO_INSTALL_EXT_PACKAGES STREQUAL MISSING)
-        set(yaml-cpp_FIND_REQUIRED FALSE)
-    endif()
-
-    include(FindPackageHandleStandardArgs)
-    find_package_handle_standard_args(yaml-cpp
-        REQUIRED_VARS
-            ${_yaml-cpp_REQUIRED_VARS}
-        VERSION_VAR
-            yaml-cpp_VERSION
-    )
-
-    mark_as_advanced(yaml-cpp_INCLUDE_DIR yaml-cpp_LIBRARY yaml-cpp_VERSION)
-endif()
-
-###############################################################################
-### Create target
-
-if (yaml-cpp_FOUND AND NOT TARGET yaml-cpp::yaml-cpp)
-    add_library(yaml-cpp::yaml-cpp UNKNOWN IMPORTED GLOBAL)
-    set_target_properties(yaml-cpp::yaml-cpp PROPERTIES
-        IMPORTED_LOCATION ${yaml-cpp_LIBRARY}
-        INTERFACE_INCLUDE_DIRECTORIES ${yaml-cpp_INCLUDE_DIR}
-    )
-
-    # Required because Installyaml-cpp.cmake creates `yaml-cpp::yaml-cpp`
-    # as an alias, and aliases get resolved in exported targets, causing the
-    # find_dependency(yaml-cpp) call in OpenColorIOConfig.cmake to fail.
-    # This can be removed once Installyaml-cpp.cmake targets yaml-cpp 0.8.
-    if (NOT TARGET yaml-cpp)
-        add_library(yaml-cpp ALIAS yaml-cpp::yaml-cpp)
-    endif ()
-endif ()
-
-if (yaml-cpp_FOUND)
-    # TODO: Remove this variable and use the `yaml-cpp::yaml-cpp` target
-    # directly when the minimum version of yaml-cpp is updated to 0.8.
-    get_target_property(YAML_CPP_INCLUDE_DIR yaml-cpp::yaml-cpp INCLUDE_DIRECTORIES)
-    set(YAML_CPP_LIBRARIES yaml-cpp::yaml-cpp)
-endif ()
diff -Naur OpenColorIO/src/apps/ocioarchive/CMakeLists.txt OPENCOLORIO_BLD/src/apps/ocioarchive/CMakeLists.txt
--- OpenColorIO/src/apps/ocioarchive/CMakeLists.txt	2023-12-20 15:33:40.584451706 -0500
+++ OPENCOLORIO_BLD/src/apps/ocioarchive/CMakeLists.txt	2023-12-20 15:33:17.080776786 -0500
@@ -21,7 +21,7 @@
     PRIVATE
         apputils
         OpenColorIO
-        MINIZIP::minizip-ng
+        MINIZIP::minizip
 )
 
 include(StripUtils)
diff -Naur OpenColorIO/src/cmake/Config.cmake.in OPENCOLORIO_BLD/src/cmake/Config.cmake.in
--- OpenColorIO/src/cmake/Config.cmake.in	2023-12-20 15:33:40.592451595 -0500
+++ OPENCOLORIO_BLD/src/cmake/Config.cmake.in	2023-12-20 15:33:17.080776786 -0500
@@ -53,8 +53,8 @@
         endif()
     endif()
 
-    if (NOT TARGET MINIZIP::minizip-ng)
-        find_dependency(minizip-ng @minizip-ng_VERSION@)
+    if (NOT TARGET MINIZIP::minizip)
+        find_dependency(minizip @minizip_VERSION@)
     endif()
 
     # Remove OCIO custom find module path.
diff -Naur OpenColorIO/src/OpenColorIO/CMakeLists.txt OPENCOLORIO_BLD/src/OpenColorIO/CMakeLists.txt
--- OpenColorIO/src/OpenColorIO/CMakeLists.txt	2023-12-20 15:33:40.576451819 -0500
+++ OPENCOLORIO_BLD/src/OpenColorIO/CMakeLists.txt	2023-12-20 15:37:00.877582518 -0500
@@ -309,8 +309,8 @@
         "$<BUILD_INTERFACE:utils::from_chars>"
         "$<BUILD_INTERFACE:utils::strings>"
         "$<BUILD_INTERFACE:xxHash>"
-        ${YAML_CPP_LIBRARIES}
-        MINIZIP::minizip-ng
+        yaml-cpp::yaml-cpp
+        MINIZIP::minizip
 )
 
 if(OCIO_USE_SIMD AND OCIO_USE_SSE2NEON AND COMPILER_SUPPORTS_SSE_WITH_SSE2NEON)
@@ -388,20 +388,20 @@
     endif()
 
 
-    # Check for minizip first since our Findminizip module sets minizip-ng_LIBRARY.
-    if (minizip_LIBRARY OR minizip-ng_COMPAT)
-        get_filename_component(_minizip-ng_LIBDIR "${minizip_LIBRARY}" DIRECTORY)
-        set(_minizip-ng_NAME "minizip")
-    elseif(minizip-ng_LIBRARY)
-        get_filename_component(_minizip-ng_LIBDIR "${minizip-ng_LIBRARY}" DIRECTORY)
-        set(_minizip-ng_NAME "minizip-ng")
+    # Check for minizip first since our Findminizip module sets minizip_LIBRARY.
+    if (minizip_LIBRARY OR minizip_COMPAT)
+        get_filename_component(_minizip_LIBDIR "${minizip_LIBRARY}" DIRECTORY)
+        set(_minizip_NAME "minizip")
+    elseif(minizip_LIBRARY)
+        get_filename_component(_minizip_LIBDIR "${minizip_LIBRARY}" DIRECTORY)
+        set(_minizip_NAME "minizip")
     endif()
 
-    if (_minizip-ng_LIBDIR)
+    if (_minizip_LIBDIR)
         # Add the path to CMake list of search paths for libraries.
-        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-L${_minizip-ng_LIBDIR}")
-        # Hide the minizip-ng symbols.
-        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-hidden-l${_minizip-ng_NAME}")
+        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-L${_minizip_LIBDIR}")
+        # Hide the minizip symbols.
+        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-hidden-l${_minizip_NAME}")
     endif()
 
     if (_OCIO_LINK_FLAGS_LIST_)
diff -Naur OpenColorIO/tests/cpu/CMakeLists.txt OPENCOLORIO_BLD/tests/cpu/CMakeLists.txt
--- OpenColorIO/tests/cpu/CMakeLists.txt	2023-12-20 15:33:40.592451595 -0500
+++ OPENCOLORIO_BLD/tests/cpu/CMakeLists.txt	2023-12-20 15:33:17.080776786 -0500
@@ -37,7 +37,7 @@
             utils::strings
             ${YAML_CPP_LIBRARIES}
             testutils
-            MINIZIP::minizip-ng
+            MINIZIP::minizip
             xxHash
     )
 
