diff -Naur libpng/scripts/pnglibconf.dfa PNG_BLD/scripts/pnglibconf.dfa
--- libpng/scripts/pnglibconf.dfa	2024-10-11 10:48:58.913083752 -0400
+++ PNG_BLD/scripts/pnglibconf.dfa	2024-10-11 10:48:27.026205664 -0400
@@ -343,7 +343,7 @@
 # The TEXT values are the defaults when writing compressed text (all forms)
 
 # Include the zlib header so that the defaults below are known
-@#  include <zlib.h>
+@#  include "zlib.h"
 
 # The '@' here means to substitute the value when pnglibconf.h is built
 setting Z_DEFAULT_COMPRESSION default @Z_DEFAULT_COMPRESSION
