diff --git a/CMakeLists.txt b/CMakeLists.txt
index 042e6dd..0d3a2ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -312,6 +312,7 @@ endif()
 
 set(BITCODE2CPP "${CMAKE_CURRENT_SOURCE_DIR}/scripts/bitcode2cpp.py")
 
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/ispcrt/cmake)
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/FixWindowsPath.cmake)
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Git.cmake)
@@ -846,6 +847,10 @@ if (NOT WIN32 AND NOT APPLE)
     list(APPEND LINK_LIBRARIES "-Wl,--start-group")
 endif()
 
+# Link against zlib
+find_package(ZLIB)
+list(APPEND LINK_LIBRARIES ${ZLIB_LIBRARIES})
+
 # Link against Clang libraries
 foreach(clangLib ${CLANG_LIBRARY_LIST})
     # Preferred way to build ISPC is to link it with individual clang static
