diff -Naur geogram/src/lib/third_party/numerics/SUPERLU/cutil.c GEOGRAM_BLD/src/lib/third_party/numerics/SUPERLU/cutil.c
--- geogram/src/lib/third_party/numerics/SUPERLU/cutil.c	2024-04-19 17:13:22.947629009 -0400
+++ GEOGRAM_BLD/src/lib/third_party/numerics/SUPERLU/cutil.c	2024-04-19 17:13:53.726931559 -0400
@@ -474,7 +474,7 @@
 
 
 
-
+int
 print_complex_vec(char *what, int n, complex *vec)
 {
     int i;
diff -Naur geogram/src/lib/third_party/numerics/SUPERLU/dutil.c GEOGRAM_BLD/src/lib/third_party/numerics/SUPERLU/dutil.c
--- geogram/src/lib/third_party/numerics/SUPERLU/dutil.c	2024-04-19 17:13:22.947629009 -0400
+++ GEOGRAM_BLD/src/lib/third_party/numerics/SUPERLU/dutil.c	2024-04-19 17:13:59.518801347 -0400
@@ -470,7 +470,7 @@
 
 
 
-
+int
 print_double_vec(char *what, int n, double *vec)
 {
     int i;
diff -Naur geogram/src/lib/third_party/numerics/SUPERLU/sutil.c GEOGRAM_BLD/src/lib/third_party/numerics/SUPERLU/sutil.c
--- geogram/src/lib/third_party/numerics/SUPERLU/sutil.c	2024-04-19 17:13:22.947629009 -0400
+++ GEOGRAM_BLD/src/lib/third_party/numerics/SUPERLU/sutil.c	2024-04-19 17:14:20.190339229 -0400
@@ -470,7 +470,7 @@
 
 
 
-
+int
 print_float_vec(char *what, int n, float *vec)
 {
     int i;
diff -Naur geogram/src/lib/third_party/numerics/SUPERLU/zutil.c GEOGRAM_BLD/src/lib/third_party/numerics/SUPERLU/zutil.c
--- geogram/src/lib/third_party/numerics/SUPERLU/zutil.c	2024-04-19 17:13:22.947629009 -0400
+++ GEOGRAM_BLD/src/lib/third_party/numerics/SUPERLU/zutil.c	2024-04-19 17:14:27.294181312 -0400
@@ -474,7 +474,7 @@
 
 
 
-
+int
 print_doublecomplex_vec(char *what, int n, doublecomplex *vec)
 {
     int i;
