diff -Naur -x .git geogram_vanilla/cmake/platforms/Linux-clang.cmake geogram/cmake/platforms/Linux-clang.cmake
--- geogram_vanilla/cmake/platforms/Linux-clang.cmake	2025-08-19 16:58:22.678922313 -0400
+++ geogram/cmake/platforms/Linux-clang.cmake	2025-08-19 16:56:01.854442739 -0400
@@ -147,3 +147,4 @@
 
 endmacro()
 
+add_link_options(-latomic)
diff -Naur -x .git geogram_vanilla/CMakeLists.txt geogram/CMakeLists.txt
--- geogram_vanilla/CMakeLists.txt	2025-08-19 16:58:22.668922350 -0400
+++ geogram/CMakeLists.txt	2025-08-19 16:55:00.798668353 -0400
@@ -93,19 +93,6 @@
 ##  set(VORPALINE_SVN_REVISION ${Vorpaline_WC_REVISION})
 endif()
 
-##############################################################################
-# RPATH (where executables find the .so / DLLs)
-# - Enables RPATH support for MACOSX
-# - Makes RPATH of dynamic libraries and executable point to the directory
-#   where libraries are installed.
-
-if(VORPALINE_BUILD_DYNAMIC)
-set(CMAKE_MACOSX_RPATH 1)
-set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-endif()
-
-set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
-
 # uninstall target
 # Needs to be created before add_subdirectory() because GLFw has
 # also an uninstall target that will be inhibited if there is
diff -Naur -x .git geogram_vanilla/src/lib/geogram/CMakeLists.txt geogram/src/lib/geogram/CMakeLists.txt
--- geogram_vanilla/src/lib/geogram/CMakeLists.txt	2025-08-19 16:58:22.773921962 -0400
+++ geogram/src/lib/geogram/CMakeLists.txt	2025-08-19 16:55:24.910579255 -0400
@@ -48,7 +48,7 @@
 
 include_directories(${PROJECT_BINARY_DIR}/src/lib)
 
-add_library(geogram ${SOURCES} $<TARGET_OBJECTS:geogram_third_party>)
+add_library(geogram SHARED ${SOURCES} $<TARGET_OBJECTS:geogram_third_party>)
 
 target_compile_features(geogram PUBLIC cxx_std_17)
 
