diff -Naur --exclude=.git tk/manifest.uuid TK_BLD/manifest.uuid
--- tk/manifest.uuid	1970-01-01 00:00:00.000000000 +0000
+++ TK_BLD/manifest.uuid	2026-09-10 17:13:18.280356000 +0000
@@ -0,0 +1 @@
+git-6dbe68819aac4a22f6f7a143133ce26c5e8d2b05
diff -Naur --exclude=.git tk/unix/Makefile.in TK_BLD/unix/Makefile.in
--- tk/unix/Makefile.in	2026-09-10 16:55:00.776870000 +0000
+++ TK_BLD/unix/Makefile.in	2026-09-10 17:13:18.283051000 +0000
@@ -285,7 +285,7 @@
 LD_SEARCH_FLAGS	= @LD_SEARCH_FLAGS@
 
 # support for embedded libraries on Darwin / Mac OS X
-DYLIB_INSTALL_DIR	= $(libdir)
+DYLIB_INSTALL_DIR	= @rpath
 
 # support for building the Aqua resource file
 TK_RSRC_FILE		= @TK_RSRC_FILE@
