diff -Naur tk/macosx/GNUmakefile TK_BLD/macosx/GNUmakefile
--- tk/macosx/GNUmakefile	2024-03-14 19:40:17.907399776 -0400
+++ TK_BLD/macosx/GNUmakefile	2024-03-14 19:36:35.471180824 -0400
@@ -145,9 +145,9 @@
 endif
 ifeq (${INSTALL_BUILD}_${EMBEDDED_BUILD}_${BUILD_STYLE},1__Deployment)
 INSTALL_TARGETS		+= html-tk
-ifneq (${INSTALL_MANPAGES},)
-INSTALL_TARGETS		+= install-doc
-endif
+#ifneq (${INSTALL_MANPAGES},)
+#INSTALL_TARGETS		+= install-doc
+#endif
 endif
 
 MAKE_VARS		+= INSTALL_ROOT INSTALL_TARGETS VERSION
diff -Naur tk/unix/Makefile.in TK_BLD/unix/Makefile.in
--- tk/unix/Makefile.in	2024-03-14 19:40:27.223241339 -0400
+++ TK_BLD/unix/Makefile.in	2024-03-14 19:37:41.702055525 -0400
@@ -574,7 +574,7 @@
 	TCL_LIBRARY=$(TCLDIR)/library; export TCL_LIBRARY; \
 	TK_LIBRARY=@TK_SRC_DIR@/library; export TK_LIBRARY;
 
-all: binaries libraries doc
+all: binaries libraries
 
 binaries: ${LIB_FILE} ${WISH_EXE}
 
@@ -701,11 +701,11 @@
 	$(SHELL_ENV) valgrind $(VALGRINDARGS) ./$(TKTEST_EXE) $(SCRIPT)
 
 INSTALL_BASE_TARGETS = install-binaries install-libraries
-INSTALL_DOC_TARGETS = install-doc
+#INSTALL_DOC_TARGETS = install-doc
 INSTALL_DEV_TARGETS = install-headers
 INSTALL_DEMO_TARGETS = install-demos
 INSTALL_EXTRA_TARGETS = @EXTRA_INSTALL@
-INSTALL_TARGETS = $(INSTALL_BASE_TARGETS) $(INSTALL_DOC_TARGETS) $(INSTALL_DEV_TARGETS) \
+INSTALL_TARGETS = $(INSTALL_BASE_TARGETS) $(INSTALL_DEV_TARGETS) \
 		$(INSTALL_DEMO_TARGETS) $(INSTALL_EXTRA_TARGETS)
 
 install: $(INSTALL_TARGETS)
diff -Naur tk/win/Makefile.in TK_BLD/win/Makefile.in
--- tk/win/Makefile.in	2024-03-14 19:40:17.915399641 -0400
+++ TK_BLD/win/Makefile.in	2024-03-14 19:39:05.968623139 -0400
@@ -409,7 +409,7 @@
 # Main targets.  The default target -- all -- builds the binaries,
 # performs any post processing on libraries or documents.
 
-all: binaries libraries doc
+all: binaries libraries
 
 binaries: @LIBRARIES@ $(WISH)
 
@@ -462,7 +462,7 @@
 	gdb ./$(WISH) --command=gdb.run
 	@$(RM) gdb.run
 
-install: all install-binaries install-libraries install-doc install-demos
+install: all install-binaries install-libraries install-demos
 
 install-binaries: binaries
 	@for i in $(LIB_INSTALL_DIR) $(BIN_INSTALL_DIR) $(PKG_INSTALL_DIR); \
diff -Naur tk/win/makefile.vc TK_BLD/win/makefile.vc
--- tk/win/makefile.vc	2024-03-14 19:40:17.915399641 -0400
+++ TK_BLD/win/makefile.vc	2024-03-14 19:39:43.011993258 -0400
@@ -335,7 +335,7 @@
 all:	  release $(CAT32)
 core:	  setup $(TKSTUBLIB) $(TKLIB)
 cwish:	  $(WISHC)
-install:  install-binaries install-libraries install-docs
+install:  install-binaries install-libraries
 !if $(SYMBOLS)
 install:    install-pdbs
 !endif
