diff -Naur --exclude=.git tk/macosx/GNUmakefile TK_BLD/macosx/GNUmakefile
--- tk/macosx/GNUmakefile	2026-09-10 16:55:00.763527000 +0000
+++ TK_BLD/macosx/GNUmakefile	2026-09-10 17:14:11.640385000 +0000
@@ -143,12 +143,6 @@
 ifeq (${EMBEDDED_BUILD},)
 INSTALL_TARGETS		+= install-private-headers install-demos
 endif
-ifeq (${INSTALL_BUILD}_${EMBEDDED_BUILD}_${BUILD_STYLE},1__Deployment)
-INSTALL_TARGETS		+= html-tk
-ifneq (${INSTALL_MANPAGES},)
-INSTALL_TARGETS		+= install-doc
-endif
-endif
 
 MAKE_VARS		+= INSTALL_ROOT INSTALL_TARGETS VERSION
 MAKE_ARGS_V		= $(foreach v,${MAKE_VARS},$v='${$v}')
diff -Naur --exclude=.git tk/unix/Makefile.in TK_BLD/unix/Makefile.in
--- tk/unix/Makefile.in	2026-09-10 16:55:00.776870000 +0000
+++ TK_BLD/unix/Makefile.in	2026-09-10 17:14:11.643687000 +0000
@@ -574,7 +574,7 @@
 	TCL_LIBRARY=$(TCLDIR)/library; export TCL_LIBRARY; \
 	TK_LIBRARY=@TK_SRC_DIR@/library; export TK_LIBRARY;
 
-all: binaries libraries doc
+all: binaries libraries
 
 binaries: ${LIB_FILE} ${WISH_EXE}
 
@@ -706,7 +706,7 @@
 INSTALL_DEMO_TARGETS = install-demos
 INSTALL_EXTRA_TARGETS = @EXTRA_INSTALL@
 INSTALL_TARGETS = $(INSTALL_BASE_TARGETS) $(INSTALL_DEV_TARGETS) \
-		$(INSTALL_DEMO_TARGETS) $(INSTALL_DOC_TARGETS) $(INSTALL_EXTRA_TARGETS)
+		$(INSTALL_DEMO_TARGETS) $(INSTALL_EXTRA_TARGETS)
 
 install: $(INSTALL_TARGETS)
 
diff -Naur --exclude=.git tk/unix/configure TK_BLD/unix/configure
--- tk/unix/configure	2026-09-10 16:55:00.777205000 +0000
+++ TK_BLD/unix/configure	2026-09-10 17:14:11.646928000 +0000
@@ -11140,7 +11140,7 @@
     includedir="${libdir}/Headers"
     PRIVATE_INCLUDE_DIR="${libdir}/PrivateHeaders"
     HTML_DIR="${libdir}/Resources/Documentation/Reference/Tk"
-    EXTRA_INSTALL="install-private-headers html-tk"
+    EXTRA_INSTALL="install-private-headers"
     EXTRA_BUILD_HTML='@ln -fs contents.htm "$(HTML_INSTALL_DIR)"/TkTOC.html'
     EXTRA_INSTALL_BINARIES='@echo "Installing Info.plist to $(LIB_INSTALL_DIR)/Resources/" && $(INSTALL_DATA_DIR) "$(LIB_INSTALL_DIR)/Resources" && $(INSTALL_DATA) Tk-Info.plist "$(LIB_INSTALL_DIR)/Resources/Info.plist"'
     EXTRA_INSTALL_BINARIES="$EXTRA_INSTALL_BINARIES"' && echo "Installing Credits.html to $(LIB_INSTALL_DIR)/Resources/" && $(INSTALL_DATA_DIR) "$(LIB_INSTALL_DIR)/Resources" && $(INSTALL_DATA) Credits.html "$(LIB_INSTALL_DIR)/Resources"'
diff -Naur --exclude=.git tk/unix/configure.in TK_BLD/unix/configure.in
--- tk/unix/configure.in	2026-09-10 16:55:00.777282000 +0000
+++ TK_BLD/unix/configure.in	2026-09-10 17:14:11.647091000 +0000
@@ -722,7 +722,7 @@
     includedir="${libdir}/Headers"
     PRIVATE_INCLUDE_DIR="${libdir}/PrivateHeaders"
     HTML_DIR="${libdir}/Resources/Documentation/Reference/Tk"
-    EXTRA_INSTALL="install-private-headers html-tk"
+    EXTRA_INSTALL="install-private-headers"
     EXTRA_BUILD_HTML='@ln -fs contents.htm "$(HTML_INSTALL_DIR)"/TkTOC.html'
     EXTRA_INSTALL_BINARIES='@echo "Installing Info.plist to $(LIB_INSTALL_DIR)/Resources/" && $(INSTALL_DATA_DIR) "$(LIB_INSTALL_DIR)/Resources" && $(INSTALL_DATA) Tk-Info.plist "$(LIB_INSTALL_DIR)/Resources/Info.plist"'
     EXTRA_INSTALL_BINARIES="$EXTRA_INSTALL_BINARIES"' && echo "Installing Credits.html to $(LIB_INSTALL_DIR)/Resources/" && $(INSTALL_DATA_DIR) "$(LIB_INSTALL_DIR)/Resources" && $(INSTALL_DATA) Credits.html "$(LIB_INSTALL_DIR)/Resources"'
diff -Naur --exclude=.git tk/win/Makefile.in TK_BLD/win/Makefile.in
--- tk/win/Makefile.in	2026-09-10 16:55:00.779409000 +0000
+++ TK_BLD/win/Makefile.in	2026-09-10 17:13:34.002003000 +0000
@@ -410,7 +410,7 @@
 # Main targets.  The default target -- all -- builds the binaries,
 # performs any post processing on libraries or documents.
 
-all: binaries libraries doc
+all: binaries libraries
 
 binaries: @LIBRARIES@ $(WISH)
 
@@ -463,7 +463,7 @@
 	gdb ./$(WISH) --command=gdb.run
 	@$(RM) gdb.run
 
-install: all install-binaries install-libraries install-doc install-demos
+install: all install-binaries install-libraries install-demos
 
 install-binaries: binaries
 	@for i in $(LIB_INSTALL_DIR) $(BIN_INSTALL_DIR) $(PKG_INSTALL_DIR); \
diff -Naur --exclude=.git tk/win/makefile.vc TK_BLD/win/makefile.vc
--- tk/win/makefile.vc	2026-09-10 16:55:00.779836000 +0000
+++ TK_BLD/win/makefile.vc	2026-09-10 17:13:34.002764000 +0000
@@ -335,7 +335,7 @@
 all:	  release $(CAT32)
 core:	  setup $(TKSTUBLIB) $(TKLIB)
 cwish:	  $(WISHC)
-install:  install-binaries install-libraries install-docs
+install:  install-binaries install-libraries
 !if $(SYMBOLS)
 install:    install-pdbs
 !endif
