diff -Naur re2c/CMakeLists.txt RE2C_BLD/CMakeLists.txt
--- re2c/CMakeLists.txt	2026-09-09 09:36:54.036421014 -0400
+++ RE2C_BLD/CMakeLists.txt	2026-09-09 09:39:52.866054493 -0400
@@ -137,7 +138,6 @@
 
 set(top_srcdir "${CMAKE_CURRENT_SOURCE_DIR}")
 set(top_builddir "${CMAKE_CURRENT_BINARY_DIR}")
-set(PYTHON "${Python3_EXECUTABLE}")
 
 configure_file(doc/manpage.rst.in doc/manpage.rst @ONLY)
 configure_file(doc/help.rst.in doc/help.rst @ONLY)
@@ -486,6 +486,7 @@
 
 if(RE2C_BUILD_TESTS)
     # tests
+    if (Python3_EXECUTABLE)
     add_custom_target(tests
         DEPENDS "${RE2C_RUN_TESTS}"
         WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
@@ -539,6 +540,7 @@
         re2c_test_ver_to_vernum
         re2c_test_argsubst
     )
+    endif (Python3_EXECUTABLE)
 endif()
 
 if (RE2C_BUILD_LIBS)
@@ -678,6 +680,8 @@
 endif()
 
 if(RE2C_BUILD_TESTS)
+    if (Python3_EXECUTABLE)
     add_custom_target(check)
     add_dependencies(check check_re2c check_libre2c)
+    endif()
 endif()
