diff -Naur qtsvg/CMakeLists.txt Qt6Svg_BLD/CMakeLists.txt
--- qtsvg/CMakeLists.txt	2024-10-11 13:57:45.155801129 -0400
+++ Qt6Svg_BLD/CMakeLists.txt	2024-10-11 13:58:46.928924859 -0400
@@ -11,6 +11,8 @@
     LANGUAGES CXX C
 )
 
+find_package(ZLIB REQUIRED)
+
 find_package(Qt6 ${PROJECT_VERSION} CONFIG REQUIRED COMPONENTS BuildInternals Core)
 find_package(Qt6 ${PROJECT_VERSION} QUIET CONFIG OPTIONAL_COMPONENTS Gui Widgets Xml)
 qt_internal_project_setup()
@@ -23,4 +25,19 @@
     find_package(Qt6 ${PROJECT_VERSION} QUIET CONFIG OPTIONAL_COMPONENTS ZlibPrivate)
 endif()
 
+# Convenience function
+function(flag_strip flag_var_name flag_value)
+	string(REPLACE "${flag_value}" " " ${flag_var_name} "${${flag_var_name}}")
+	get_property(help_text CACHE ${flag_var_name} PROPERTY HELPSTRING)
+	set(${flag_var_name} "${${flag_var_name}}" CACHE STRING "${help_text}" FORCE)
+endfunction(flag_strip flag_var_name)
+
+# The /RTC1 flag is present in the default DEBUG flags list and contradicts -O2
+set(flag_value "/RTC1")
+set(configs RELWITHDEBINFO DEBUG)
+foreach(config ${configs})
+	set(flag_var_name "CMAKE_CXX_FLAGS_${config}")
+	flag_strip(${flag_var_name} ${flag_value})
+endforeach(config ${configs})
+
 qt_build_repo()
diff -Naur qtsvg/src/svg/CMakeLists.txt Qt6Svg_BLD/src/svg/CMakeLists.txt
--- qtsvg/src/svg/CMakeLists.txt	2024-10-11 13:57:45.158801086 -0400
+++ Qt6Svg_BLD/src/svg/CMakeLists.txt	2024-10-11 13:59:36.937211035 -0400
@@ -37,6 +37,7 @@
     PUBLIC_LIBRARIES
         Qt::Core
         Qt::Gui
+        ZLIB::ZLIB
     PRIVATE_MODULE_INTERFACE
         Qt::CorePrivate
         Qt::GuiPrivate
