diff -Naur qtsvg/CMakeLists.txt Qt6Svg_BLD/CMakeLists.txt
--- qtsvg/CMakeLists.txt	2023-12-19 20:56:21.873344047 -0500
+++ Qt6Svg_BLD/CMakeLists.txt	2023-12-20 10:28:48.591993474 -0500
@@ -11,6 +11,8 @@
     LANGUAGES CXX C
 )
 
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake")
+
 find_package(Qt6 ${PROJECT_VERSION} CONFIG REQUIRED COMPONENTS BuildInternals)
 
 # This should be called as early as possible, just after find_package(BuildInternals) where it is
@@ -28,4 +30,19 @@
     find_package(Qt6 ${PROJECT_VERSION} QUIET CONFIG OPTIONAL_COMPONENTS ZlibPrivate)
 endif()
 
+# Convenience function
+function(flag_strip flag_var_name flag_value)
+	string(REPLACE "${flag_value}" " " ${flag_var_name} "${${flag_var_name}}")
+	get_property(help_text CACHE ${flag_var_name} PROPERTY HELPSTRING)
+	set(${flag_var_name} "${${flag_var_name}}" CACHE STRING "${help_text}" FORCE)
+endfunction(flag_strip flag_var_name)
+
+# The /RTC1 flag is present in the default DEBUG flags list and contradicts -O2
+set(flag_value "/RTC1")
+set(configs RELWITHDEBINFO DEBUG)
+foreach(config ${configs})
+	set(flag_var_name "CMAKE_CXX_FLAGS_${config}")
+	flag_strip(${flag_var_name} ${flag_value})
+endforeach(config ${configs})
+
 qt_build_repo()
diff -Naur qtsvg/src/svg/CMakeLists.txt Qt6Svg_BLD/src/svg/CMakeLists.txt
--- qtsvg/src/svg/CMakeLists.txt	2023-12-19 20:56:21.877344964 -0500
+++ Qt6Svg_BLD/src/svg/CMakeLists.txt	2023-12-20 10:19:52.104033439 -0500
@@ -12,6 +12,8 @@
     qt_find_package(WrapZLIB PROVIDED_TARGETS WrapZLIB::WrapZLIB)
 endif()
 
+find_package(ZLIB)
+
 qt_internal_add_module(Svg
     SOURCES
         qsvgfont.cpp qsvgfont_p.h
@@ -43,6 +45,7 @@
     PUBLIC_LIBRARIES
         Qt::Core
         Qt::Gui
+        ${ZLIB_LIBRARY}
     PRIVATE_MODULE_INTERFACE
         Qt::CorePrivate
         Qt::GuiPrivate
