diff -Naur opencv/.cache/.gitignore OPENCV_BLD/.cache/.gitignore
--- opencv/.cache/.gitignore	1969-12-31 19:00:00.000000000 -0500
+++ OPENCV_BLD/.cache/.gitignore	2024-03-06 11:14:00.970668988 -0500
@@ -0,0 +1 @@
+*
diff -Naur opencv/cmake/OpenCVInstallLayout.cmake OPENCV_BLD/cmake/OpenCVInstallLayout.cmake
--- opencv/cmake/OpenCVInstallLayout.cmake	2024-03-06 11:15:18.249391428 -0500
+++ OPENCV_BLD/cmake/OpenCVInstallLayout.cmake	2024-03-06 11:14:00.642674496 -0500
@@ -22,12 +22,12 @@
 
 elseif(WIN32 AND CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
 
-  if(DEFINED OpenCV_RUNTIME AND DEFINED OpenCV_ARCH)
-    ocv_update(OPENCV_INSTALL_BINARIES_PREFIX "${OpenCV_ARCH}/${OpenCV_RUNTIME}/")
-  else()
-    message(STATUS "Can't detect runtime and/or arch")
+	#if(DEFINED OpenCV_RUNTIME AND DEFINED OpenCV_ARCH)
+	#ocv_update(OPENCV_INSTALL_BINARIES_PREFIX "${OpenCV_ARCH}/${OpenCV_RUNTIME}/")
+	#else()
+	#message(STATUS "Can't detect runtime and/or arch")
     ocv_update(OPENCV_INSTALL_BINARIES_PREFIX "")
-  endif()
+	#endif()
   if(OpenCV_STATIC)
     ocv_update(OPENCV_INSTALL_BINARIES_SUFFIX "staticlib")
   else()
diff -Naur opencv/CMakeLists.txt OPENCV_BLD/CMakeLists.txt
--- opencv/CMakeLists.txt	2024-06-25 10:25:05.539691869 -0400
+++ OPENCV_BLD/CMakeLists.txt	2024-06-25 10:28:39.828624534 -0400
@@ -13,6 +13,8 @@
 ")
 endif()
 
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
 include(cmake/OpenCVMinDepVersions.cmake)
 
 if(CMAKE_SYSTEM_NAME MATCHES WindowsPhone OR CMAKE_SYSTEM_NAME MATCHES WindowsStore)
@@ -68,6 +70,10 @@
   cmake_policy(SET CMP0071 NEW)  # CMake 3.10+: Let `AUTOMOC` and `AUTOUIC` process `GENERATED` files.
 endif()
 
+if (POLICY CMP0074)
+  cmake_policy(SET CMP0074 NEW)  # CMake 3.12+: find_package() uses <PackageName>_ROOT variables.
+endif (POLICY CMP0074)
+
 if(POLICY CMP0075)
   cmake_policy(SET CMP0075 NEW)  # CMake 3.12+: Include file check macros honor `CMAKE_REQUIRED_LIBRARIES`
 endif()
@@ -1154,7 +1160,7 @@
 if(ANDROID OR NOT UNIX)
   install(FILES ${OPENCV_LICENSE_FILE}
         PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
-        DESTINATION ./ COMPONENT libs)
+        DESTINATION doc/legal/other/opencv COMPONENT libs)
   if(OPENCV_README_FILE)
     install(FILES ${OPENCV_README_FILE}
             PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
