diff --git a/libs/iostreams/CMakeLists.txt b/libs/iostreams/CMakeLists.txt
index 1d8352f5..11f8e584 100644
--- a/libs/iostreams/CMakeLists.txt
+++ b/libs/iostreams/CMakeLists.txt
@@ -6,6 +6,8 @@ cmake_minimum_required(VERSION 3.5...3.16)
 
 project(boost_iostreams VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
 add_library(boost_iostreams
   src/file_descriptor.cpp
   src/mapped_file.cpp
diff --git a/libs/locale/CMakeLists.txt b/libs/locale/CMakeLists.txt
index f47ad394..21956d88 100644
--- a/libs/locale/CMakeLists.txt
+++ b/libs/locale/CMakeLists.txt
@@ -6,6 +6,8 @@ cmake_minimum_required(VERSION 3.11...3.20)
 
 project(boost_locale VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
 # Using glob here is ok as it is only for headers
 file(GLOB_RECURSE headers include/*.hpp)
 
