diff -Naur boost/libs/iostreams/CMakeLists.txt BOOST_BLD/libs/iostreams/CMakeLists.txt
--- boost/libs/iostreams/CMakeLists.txt	2023-09-19 20:13:32.123188586 -0400
+++ BOOST_BLD/libs/iostreams/CMakeLists.txt	2023-09-19 20:07:09.587847180 -0400
@@ -6,6 +6,8 @@
 
 project(boost_iostreams VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
 add_library(boost_iostreams
   src/file_descriptor.cpp
   src/mapped_file.cpp
diff -Naur boost/libs/locale/CMakeLists.txt BOOST_BLD/libs/locale/CMakeLists.txt
--- boost/libs/locale/CMakeLists.txt	2023-09-19 20:13:32.379183041 -0400
+++ BOOST_BLD/libs/locale/CMakeLists.txt	2023-09-19 20:12:31.660513836 -0400
@@ -6,6 +6,8 @@
 
 project(boost_locale VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
 # Using glob here is ok as it is only for headers
 file(GLOB_RECURSE headers include/*.hpp)
 
