--- a/frmts/png/pngdataset.cpp
+++ b/frmts/png/pngdataset.cpp
@@ -42,7 +42,9 @@
 #pragma clang diagnostic pop
 #endif
 
-#include <csetjmp>
+extern "C" {
+#include <setjmp.h>
+}
 
 #include <algorithm>
 #include <limits>
--- a/frmts/png/pngdataset.h
+++ b/frmts/png/pngdataset.h
@@ -40,7 +40,9 @@
 
 #include "png.h"
 
-#include <csetjmp>
+extern "C" {
+#include <setjmp.h>
+}
 
 #include <algorithm>
 #include <array>
