diff -Naur gdal/cmake/modules/packages/FindSQLite3.cmake GDAL_BLD/cmake/modules/packages/FindSQLite3.cmake
--- gdal/cmake/modules/packages/FindSQLite3.cmake	2024-06-25 09:56:45.835498631 -0400
+++ GDAL_BLD/cmake/modules/packages/FindSQLite3.cmake	2024-06-25 09:59:46.365399374 -0400
@@ -57,7 +57,7 @@
             NAMES  sqlite3.h
             HINTS ${PC_SQLITE3_INCLUDE_DIRS})
   find_library(SQLite3_LIBRARY
-               NAMES sqlite3 sqlite3_i
+               NAMES sqlite3 sqlite3_i libsqlite3 libsqlite libsqlite3_i
                NAMES_PER_DIR
                HINTS ${PC_SQLITE3_LIBRARY_DIRS})
 endif()
 
 # Extract version information from the header file
diff -Naur gdal/gcore/gdal_priv.h GDAL_BLD/gcore/gdal_priv.h
--- gdal/gcore/gdal_priv.h	2024-06-26 17:01:06.199711566 -0400
+++ GDAL_BLD/gcore/gdal_priv.h	2024-06-26 17:00:48.433037793 -0400
@@ -96,6 +96,13 @@
 /*                       GDALMultiDomainMetadata                        */
 /************************************************************************/
 
+#if defined(_WIN32)
+#define STRCASECMP(a, b) (_stricmp(a, b))
+#else
+/** Alias for strcasecmp() */
+#define STRCASECMP(a, b) (strcasecmp(a, b))
+#endif
+
 //! @cond Doxygen_Suppress
 class CPL_DLL GDALMultiDomainMetadata
 {
diff -Naur gdal/gdal.cmake GDAL_BLD/gdal.cmake
--- gdal/gdal.cmake	2024-06-25 09:56:45.660500506 -0400
+++ GDAL_BLD/gdal.cmake	2024-06-25 09:55:37.834206511 -0400
@@ -379,12 +379,6 @@
 set_property(TARGET ${GDAL_LIB_TARGET_NAME} PROPERTY PLUGIN_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/gdalplugins")
 file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/gdalplugins")
 
-if (MSVC)
-  set(GDAL_DEBUG_POSTFIX
-      "d"
-      CACHE STRING "Postfix to add to the GDAL dll name for debug builds")
-  set_target_properties(${GDAL_LIB_TARGET_NAME} PROPERTIES DEBUG_POSTFIX "${GDAL_DEBUG_POSTFIX}")
-endif ()
 if (MINGW AND BUILD_SHARED_LIBS)
     set_target_properties(${GDAL_LIB_TARGET_NAME} PROPERTIES SUFFIX "-${GDAL_SOVERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
 endif ()
diff -Naur gdal/port/cpl_port.h GDAL_BLD/port/cpl_port.h
--- gdal/port/cpl_port.h	2024-06-25 09:56:45.954497348 -0400
+++ GDAL_BLD/port/cpl_port.h	2024-06-25 09:55:37.836206491 -0400
@@ -1123,7 +1123,7 @@
 #define TRUE 1
 #endif
 
-#if __clang_major__ >= 4 || (__clang_major__ == 3 && __clang_minor__ >= 8)
+#if defined(__clang_major__) && (__clang_major__ >= 4 || ( __clang_major__ == 3 && __clang_minor__ >= 8) )
 #define CPL_NOSANITIZE_UNSIGNED_INT_OVERFLOW                                   \
     __attribute__((no_sanitize("unsigned-integer-overflow")))
 #else
