diff -Naur PROJ/CMakeLists.txt PROJ_BLD/CMakeLists.txt
--- PROJ/CMakeLists.txt	2024-03-06 12:18:36.881778166 -0500
+++ PROJ_BLD/CMakeLists.txt	2024-03-06 12:18:22.678018313 -0500
@@ -197,7 +197,7 @@
 # Check for sqlite3
 ################################################################################
 
-find_program(EXE_SQLITE3 sqlite3)
+find_program(EXE_SQLITE3 sqlite3 HINTS ${SQLite3_ROOT}/bin)
 if(NOT EXE_SQLITE3)
   message(SEND_ERROR "sqlite3 binary not found!")
 endif()
