diff -Naur itk/generic/itk_cmds.c ITK_BLD/generic/itk_cmds.c
--- itk/generic/itk_cmds.c	2024-07-10 10:44:13.330651607 -0400
+++ ITK_BLD/generic/itk_cmds.c	2024-07-10 10:32:07.734264686 -0400
@@ -111,8 +111,7 @@
  * ------------------------------------------------------------------------
  */
 static int
-Initialize(interp)
-    Tcl_Interp *interp;  /* interpreter to be updated */
+Initialize(Tcl_Interp *interp)
 {
     Tcl_Namespace *itkNs, *parserNs;
     ClientData parserInfo;
@@ -256,8 +255,7 @@
  * ------------------------------------------------------------------------
  */
 int
-Itk_Init(interp)
-    Tcl_Interp *interp;  /* interpreter to be updated */
+Itk_Init(Tcl_Interp *interp)
 {
     if (Initialize(interp) != TCL_OK) {
 	return TCL_ERROR;
@@ -282,8 +280,7 @@
  * ------------------------------------------------------------------------
  */  
 int 
-Itk_SafeInit(interp)
-    Tcl_Interp *interp;  /* interpreter to be updated */ 
+Itk_SafeInit(Tcl_Interp *interp)
 {   
     if (Initialize(interp) != TCL_OK) {
         return TCL_ERROR;
@@ -316,11 +313,7 @@
  */
 /* ARGSUSED */
 int
-Itk_ConfigBodyCmd(dummy, interp, objc, objv)
-    ClientData dummy;        /* unused */
-    Tcl_Interp *interp;      /* current interpreter */
-    int objc;                /* number of arguments */
-    Tcl_Obj *CONST objv[];   /* argument objects */
+Itk_ConfigBodyCmd(ClientData dummy, Tcl_Interp *interp, int objc, Tcl_Obj *CONST objv[])
 {
     int result = TCL_OK;
 
diff -Naur itk/win/rc/itk.rc ITK_BLD/win/rc/itk.rc
--- itk/win/rc/itk.rc	2024-07-10 10:44:13.334651538 -0400
+++ ITK_BLD/win/rc/itk.rc	2024-07-10 10:32:07.734264686 -0400
@@ -1,59 +1,59 @@
-//
-// Version resource script.
-//
-
-#include <winver.h>
-#include <itcl.h>
-#include <itk.h>
-
-//
-// build-up the name suffix that defines the type of build this is.
-//
-#if DEBUG
-#define SUFFIX_DEBUG	    "g"
-#else
-#define SUFFIX_DEBUG	    ""
-#endif
-
-#define SUFFIX		    SUFFIX_DEBUG
-
-
-VS_VERSION_INFO VERSIONINFO
- FILEVERSION    ITK_MAJOR_VERSION,ITK_MINOR_VERSION,ITK_RELEASE_LEVEL,ITK_RELEASE_SERIAL
- PRODUCTVERSION ITK_MAJOR_VERSION,ITK_MINOR_VERSION,ITK_RELEASE_LEVEL,ITK_RELEASE_SERIAL
- FILEFLAGSMASK	0x3fL
-#if DEBUG
- FILEFLAGS	VS_FF_DEBUG
-#else
- FILEFLAGS	0x0L
-#endif
- FILEOS		VOS__WINDOWS32
- FILETYPE	VFT_DLL
- FILESUBTYPE	0x0L
-
-BEGIN
-    BLOCK "StringFileInfo"
-    BEGIN
-        BLOCK "040904b0"
-        BEGIN
-            VALUE "FileDescription", "[Incr Tk] Object-Oriented Tk extension\0"
-            VALUE "Authors", "Michael McLennan\0"
-            VALUE "OriginalFilename", "itk" STRINGIFY(ITK_MAJOR_VERSION) STRINGIFY(ITK_MINOR_VERSION) SUFFIX ".dll\0"
-            VALUE "CompanyName", "Bell Labs Innovations for Lucent Technologies\0"
-            VALUE "FileVersion", ITK_PATCH_LEVEL
-            VALUE "LegalCopyright", "Copyright \251 1993-2003\0"
-            VALUE "ProductName", "[Incr Tk] " ITK_VERSION " for Windows\0"
-            VALUE "ProductVersion", ITK_PATCH_LEVEL
-        END
-    END
-    BLOCK "VarFileInfo"
-    BEGIN
-        VALUE "Translation", 0x409, 1200
-    END
-END
-
-//
-// Icons
-//
-
-toaster		ICON	DISCARDABLE	"itk.ico"
+//
+// Version resource script.
+//
+
+#include <winver.h>
+#include <itcl.h>
+#include <itk.h>
+
+//
+// build-up the name suffix that defines the type of build this is.
+//
+#if DEBUG
+#define SUFFIX_DEBUG	    "g"
+#else
+#define SUFFIX_DEBUG	    ""
+#endif
+
+#define SUFFIX		    SUFFIX_DEBUG
+
+
+VS_VERSION_INFO VERSIONINFO
+ FILEVERSION    ITK_MAJOR_VERSION,ITK_MINOR_VERSION,ITK_RELEASE_LEVEL,ITK_RELEASE_SERIAL
+ PRODUCTVERSION ITK_MAJOR_VERSION,ITK_MINOR_VERSION,ITK_RELEASE_LEVEL,ITK_RELEASE_SERIAL
+ FILEFLAGSMASK	0x3fL
+#if DEBUG
+ FILEFLAGS	VS_FF_DEBUG
+#else
+ FILEFLAGS	0x0L
+#endif
+ FILEOS		VOS__WINDOWS32
+ FILETYPE	VFT_DLL
+ FILESUBTYPE	0x0L
+
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "040904b0"
+        BEGIN
+            VALUE "FileDescription", "[Incr Tk] Object-Oriented Tk extension\0"
+            VALUE "Authors", "Michael McLennan\0"
+            VALUE "OriginalFilename", "itk" STRINGIFY(ITK_MAJOR_VERSION) STRINGIFY(ITK_MINOR_VERSION) SUFFIX ".dll\0"
+            VALUE "CompanyName", "Bell Labs Innovations for Lucent Technologies\0"
+            VALUE "FileVersion", ITK_PATCH_LEVEL
+            VALUE "LegalCopyright", "Copyright \251 1993-2003\0"
+            VALUE "ProductName", "[Incr Tk] " ITK_VERSION " for Windows\0"
+            VALUE "ProductVersion", ITK_PATCH_LEVEL
+        END
+    END
+    BLOCK "VarFileInfo"
+    BEGIN
+        VALUE "Translation", 0x409, 1200
+    END
+END
+
+//
+// Icons
+//
+
+toaster		ICON	DISCARDABLE	"itk.ico"
