diff -Naur OpenMesh/cmake-library/VCI/VCICommon.cmake OPENMESH_BLD/cmake-library/VCI/VCICommon.cmake
--- OpenMesh/cmake-library/VCI/VCICommon.cmake	2023-12-21 19:31:59.779342577 -0500
+++ OPENMESH_BLD/cmake-library/VCI/VCICommon.cmake	2023-12-21 13:23:27.538654820 -0500
@@ -41,7 +41,7 @@
 if (CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
   set (VCI_PROJECT_DATADIR ".")
   set (VCI_PROJECT_LIBDIR "lib")
-  set (VCI_PROJECT_BINDIR ".")
+  set (VCI_PROJECT_BINDIR "bin")
   set (VCI_PROJECT_PLUGINDIR "Plugins")
   if (NOT EXISTS ${CMAKE_BINARY_DIR}/Build/${VCI_PROJECT_LIBDIR})
     file (MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/Build/${VCI_PROJECT_LIBDIR})
diff -Naur OpenMesh/CMakeLists.txt OPENMESH_BLD/CMakeLists.txt
--- OpenMesh/CMakeLists.txt	2023-12-21 19:31:59.735343328 -0500
+++ OPENMESH_BLD/CMakeLists.txt	2023-12-21 19:32:10.127166493 -0500
@@ -2,8 +2,8 @@
 
 
 # Set and enforce C++-11 flags
-#set( CMAKE_CXX_STANDARD_REQUIRED TRUE )
-#set( CMAKE_CXX_STANDARD 11 )
+set( CMAKE_CXX_STANDARD_REQUIRED TRUE )
+set( CMAKE_CXX_STANDARD 17 )
 
 enable_testing()
 
@@ -38,7 +38,6 @@
 
 # add our macro directory to cmake search path
 set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${CMAKE_CURRENT_SOURCE_DIR}/cmake-library/finders ${CMAKE_CURRENT_SOURCE_DIR}/cmake-library/VCI )
-set (CMAKE_DEBUG_POSTFIX "d")
 
 # include our cmake files
 include (VCICommon)
diff -Naur OpenMesh/src/OpenMesh/Core/CMakeLists.txt OPENMESH_BLD/src/OpenMesh/Core/CMakeLists.txt
--- OpenMesh/src/OpenMesh/Core/CMakeLists.txt	2023-12-21 19:31:59.807342103 -0500
+++ OPENMESH_BLD/src/OpenMesh/Core/CMakeLists.txt	2023-12-21 13:23:27.542654752 -0500
@@ -184,7 +184,7 @@
     FILE(GLOB files_install_libs "${CMAKE_BINARY_DIR}/Build/lib/*.lib" )
     FILE(GLOB files_install_dlls "${CMAKE_BINARY_DIR}/Build/*.dll" )
     INSTALL(FILES ${files_install_libs} DESTINATION lib )
-    INSTALL(FILES ${files_install_dlls} DESTINATION . )
+    INSTALL(FILES ${files_install_dlls} DESTINATION bin )
   endif()
 endif()
 
