diff -Naur openexr/CMakeLists.txt OPENEXR_BLD/CMakeLists.txt
--- openexr/CMakeLists.txt
+++ OPENEXR_BLD/CMakeLists.txt
@@ -163,7 +163,7 @@
   add_subdirectory(website)
 endif()
 
-if (OPENEXR_BUILD_LIBS AND NOT OPENEXR_IS_SUBPROJECT)
+if (OPENEXR_BUILD_LIBS AND OPENEXR_BUILD_EXAMPLES AND NOT OPENEXR_IS_SUBPROJECT)
   # Even if not building the website, still make sure the website example code compiles.
   add_subdirectory(website/src)
 endif()
